--- ff-orig/mozilla-28.0/dom/base/nsJSEnvironment.h	2014-03-15 00:19:12.000000000 -0500
+++ ff-dev/mozilla-28.0/dom/base/nsJSEnvironment.h	2015-01-25 12:15:21.155403005 -0600
@@ -41,6 +41,18 @@
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS_AMBIGUOUS(nsJSContext,
                                                          nsIScriptContext)
 
+#ifdef TxJS
+  virtual nsresult EvaluateString(const nsAString& aScript,
+                                  JSObject* aScopeObject,
+                                  nsIPrincipal *principal,
+                                  nsIPrincipal *originPrincipal,
+                                  const char *aURL,
+                                  uint32_t aLineNo,
+                                  JSVersion aVersion,
+                                  nsAString *aRetValue,
+                                  bool* aIsUndefined,
+                                  const JS::Value &ispect);
+#endif
   virtual nsresult EvaluateString(const nsAString& aScript,
                                   JS::Handle<JSObject*> aScopeObject,
                                   JS::CompileOptions &aOptions,

