--- ff-orig/mozilla-28.0/js/src/vm/GlobalObject.cpp	2014-03-15 00:19:23.000000000 -0500
+++ ff-dev/mozilla-28.0/js/src/vm/GlobalObject.cpp	2015-01-25 14:38:19.511152189 -0600
@@ -13,6 +13,10 @@
 #include "jsmath.h"
 #include "json.h"
 #include "jsweakmap.h"
+#ifdef TxJS
+#include "jstransaction.h"
+#include "jsjam.h"
+#endif
 
 #include "builtin/Eval.h"
 #if EXPOSE_INTL_API
@@ -520,6 +524,10 @@
 #if ENABLE_PARALLEL_JS
            js_InitParallelArrayClass(cx, global) &&
 #endif
+#ifdef TxJS
+           js_InitTxClass(cx, global) &&
+           js_InitJAMClass(cx, global) &&
+#endif
            true;
 }
 

