--- ff-orig/mozilla-28.0/js/src/jscntxtinlines.h	2014-03-15 00:19:20.000000000 -0500
+++ ff-dev/mozilla-28.0/js/src/jscntxtinlines.h	2015-01-25 12:15:21.523402995 -0600
@@ -216,10 +216,14 @@
 #ifdef DEBUG
     bool alreadyThrowing = cx->isExceptionPending();
 #endif
+#ifndef TxJS
     assertSameCompartment(cx, args);
+#endif
     bool ok = native(cx, args.length(), args.base());
     if (ok) {
+//#ifndef TxJS
         assertSameCompartment(cx, args.rval());
+//#endif
         JS_ASSERT_IF(!alreadyThrowing, !cx->isExceptionPending());
     }
     return ok;

