--- ff-orig/mozilla-28.0/js/src/frontend/FullParseHandler.h	2014-03-15 00:19:19.000000000 -0500
+++ ff-dev/mozilla-28.0/js/src/frontend/FullParseHandler.h	2015-01-25 12:15:21.295403001 -0600
@@ -279,8 +279,18 @@
     void addStatementToList(ParseNode *list, ParseNode *stmt, PC *pc) {
         JS_ASSERT(list->isKind(PNK_STATEMENTLIST));
 
+#ifdef TxJS_HOIST
+        if (stmt->isKind(PNK_TX) && stmt->pn_xflags & PNX_FUNCDEFS) {
+          stmt->pn_xflags &= ~PNX_FUNCDEFS;
+          list->pn_xflags |= PNX_FUNCDEFS;
+        }
+#endif
         if (stmt->isKind(PNK_FUNCTION)) {
+#ifdef TxJS_HOIST
+            if (pc->atBodyLevel() || pc->hasTxTop) {
+#else
             if (pc->atBodyLevel()) {
+#endif
                 // PNX_FUNCDEFS notifies the emitter that the block contains
                 // body-level function definitions that should be processed
                 // before the rest of nodes.
@@ -289,6 +299,11 @@
                 // General deoptimization was done in Parser::functionDef.
                 JS_ASSERT_IF(pc->sc->isFunctionBox(),
                              pc->sc->asFunctionBox()->hasExtensibleScope());
+#ifdef TxJS
+                // Communicate to |txStmt| that there is a function
+                // definition within.
+                pc->hasFunctionStmt = true;
+#endif
             }
         }
 

