--- ff-orig/mozilla-28.0/js/src/frontend/TokenStream.cpp	2014-03-15 00:19:19.000000000 -0500
+++ ff-dev/mozilla-28.0/js/src/frontend/TokenStream.cpp	2015-01-25 12:15:21.303403001 -0600
@@ -1816,6 +1816,9 @@
       case TOK_MULASSIGN:       return "TOK_MULASSIGN";
       case TOK_DIVASSIGN:       return "TOK_DIVASSIGN";
       case TOK_MODASSIGN:       return "TOK_MODASSIGN";
+#ifdef TxJS
+      case TOK_TX:              return "TOK_TX";
+#endif
       case TOK_EXPORT:          return "TOK_EXPORT";
       case TOK_IMPORT:          return "TOK_IMPORT";
       case TOK_LIMIT:           break;

