--- ff-orig/mozilla-28.0/js/src/jsapi.h	2014-03-15 00:19:20.000000000 -0500
+++ ff-dev/mozilla-28.0/js/src/jsapi.h	2015-01-25 12:15:21.435402997 -0600
@@ -963,7 +963,14 @@
  */
 #define JSFUN_GENERIC_NATIVE   0x800
 
+#ifdef TxJS
+#define JSFUN_TX_NOSUSPEND     0x1000   /* indicates that a call to this
+                                           function should not suspend
+                                           an active transaction */
+#define JSFUN_FLAGS_MASK       0x1e00   /* | of all the JSFUN_* flags */
+#else
 #define JSFUN_FLAGS_MASK       0xe00    /* | of all the JSFUN_* flags */
+#endif
 
 /*
  * The first call to JS_CallOnce by any thread in a process will call 'func'.
@@ -996,6 +1003,11 @@
 extern JS_PUBLIC_API(JSString *)
 JS_GetEmptyString(JSRuntime *rt);
 
+#ifdef TxJS
+extern JS_PUBLIC_API(jsval)
+JS_GetIntrospector(JSContext *cx);
+#endif
+
 /*
  * Format is a string of the following characters (spaces are insignificant),
  * specifying the tabulated type conversions:
@@ -3152,6 +3164,7 @@
  * Security protocol.
  */
 struct JSPrincipals {
+
     /* Don't call "destroy"; use reference counting macros below. */
 #ifdef JS_THREADSAFE
     mozilla::Atomic<int32_t> refcount;
@@ -3460,6 +3473,9 @@
     bool forEval;
     bool noScriptRval;
     bool selfHostingMode;
+#ifdef TxJS
+    Value introspector;
+#endif
     bool canLazilyParse;
     bool strictOption;
     bool extraWarningsOption;

