--- ff-orig/mozilla-28.0/js/src/vm/Interpreter.h	2014-03-15 00:19:23.000000000 -0500
+++ ff-dev/mozilla-28.0/js/src/vm/Interpreter.h	2015-01-25 12:15:21.703402989 -0600
@@ -320,6 +320,20 @@
 extern void
 UnwindScope(JSContext *cx, AbstractFramePtr frame, uint32_t stackDepth);
 
+#ifdef TxJS
+extern bool 
+BeginTx(JSContext *cx, const js::Value &tx_handler);
+
+extern bool 
+IntrospectTx(JSContext *cx, TxContext *tx, InterpreterActivation *activation, bool isEnd, HandleScript script, int *path, bool *newType);
+
+extern bool
+TxSubsumed(JSContext *cx, const js::Value &ispect);
+
+extern bool
+SuspendCall(JSContext *cx, TxContext *tx, CallArgs &args, bool *construct);
+#endif
+
 /*
  * Unwind for an uncatchable exception. This means not running finalizers, etc;
  * just preserving the basic engine stack invariants.

