--- ff-orig/mozilla-28.0/js/src/jsopcode.cpp	2014-03-15 00:19:20.000000000 -0500
+++ ff-dev/mozilla-28.0/js/src/jsopcode.cpp	2015-01-25 12:15:21.567402993 -0600
@@ -150,8 +150,17 @@
         return NumBlockSlots(script, pc) + 2;
       default:
         /* stack: fun, this, [argc arguments] */
+#ifdef TxJS
+        JS_ASSERT(op == JSOP_NEW || op == JSOP_CALL || op == JSOP_EVAL ||
+                  op == JSOP_BEGIN_TX || op == JSOP_FUNCALL || op == JSOP_FUNAPPLY);
+        // There's no fun or this for transaction entry.
+        if (op == JSOP_BEGIN_TX)
+            return GET_ARGC(pc);
+        else
+#else
         JS_ASSERT(op == JSOP_NEW || op == JSOP_CALL || op == JSOP_EVAL ||
                   op == JSOP_FUNCALL || op == JSOP_FUNAPPLY);
+#endif
         return 2 + GET_ARGC(pc);
     }
 }

