--- ff-orig/mozilla-19.0.2/js/src/Makefile.in	2013-03-07 04:48:46.000000000 -0600
+++ ff-dev/mozilla-19.0.2/js/src/Makefile.in	2015-01-19 17:33:09.999152205 -0600
@@ -86,6 +86,7 @@
 		jsinfer.cpp \
 		jsinterp.cpp \
 		jsiter.cpp \
+    jsjam.cpp \
 		jslog2.cpp \
 		jsmath.cpp \
 		jsmemorymetrics.cpp \
@@ -104,6 +105,9 @@
 		jsscope.cpp \
 		jsscript.cpp \
 		jsstr.cpp \
+    jstransaction.cpp \
+    jstx.cpp \
+    jstxaction.cpp \
 		jstypedarray.cpp \
 		jsutil.cpp \
 		jswatchpoint.cpp \

