--- ff-orig/mozilla-19.0.2/js/src/jscntxtinlines.h	2013-03-07 04:48:48.000000000 -0600
+++ ff-dev/mozilla-19.0.2/js/src/jscntxtinlines.h	2015-01-19 17:33:09.847152210 -0600
@@ -360,10 +360,14 @@
 #ifdef DEBUG
     bool alreadyThrowing = cx->isExceptionPending();
 #endif
+#ifndef TxJS
     assertSameCompartment(cx, args);
+#endif
     bool ok = native(cx, args.length(), args.base());
     if (ok) {
+//#ifndef TxJS
         assertSameCompartment(cx, args.rval());
+//#endif
         JS_ASSERT_IF(!alreadyThrowing, !cx->isExceptionPending());
     }
     return ok;

