--- ff-orig/mozilla-19.0.2/js/src/frontend/TokenStream.cpp	2013-03-07 04:48:46.000000000 -0600
+++ ff-dev/mozilla-19.0.2/js/src/frontend/TokenStream.cpp	2015-01-19 17:33:09.691152214 -0600
@@ -2309,6 +2309,9 @@
       case TOK_MULASSIGN:       return "TOK_MULASSIGN";
       case TOK_DIVASSIGN:       return "TOK_DIVASSIGN";
       case TOK_MODASSIGN:       return "TOK_MODASSIGN";
+#ifdef TxJS
+      case TOK_TX:              return "TOK_TX";
+#endif
       case TOK_EXPORT:          return "TOK_EXPORT";
       case TOK_IMPORT:          return "TOK_IMPORT";
       case TOK_LIMIT:           break;

