--- ff-orig/mozilla-19.0.2/js/src/vm/Stack-inl.h	2013-03-07 04:48:51.000000000 -0600
+++ ff-dev/mozilla-19.0.2/js/src/vm/Stack-inl.h	2015-01-19 17:33:10.075152203 -0600
@@ -118,6 +118,9 @@
 
     if (prevfp->hasPushedSPSFrame())
         setPushedSPSFrame();
+#ifdef TxJS
+    suspendedTx_ = NULL;
+#endif
 }
 
 inline void
@@ -152,6 +155,9 @@
     JS_ASSERT(!hasBlockChain());
     JS_ASSERT(!hasHookData());
     JS_ASSERT(annotation() == NULL);
+#ifdef TxJS
+    suspendedTx_ = NULL;
+#endif
 
     initVarsToUndefined();
 }
@@ -173,6 +179,9 @@
     prev_ = prev;
     ncode_ = ncode;
     u.nactual = nactual;
+#ifdef TxJS
+    suspendedTx_ = NULL;
+#endif
 }
 
 inline bool

