--- ff-orig/mozilla-27.0.1/content/base/src/nsFrameLoader.h	2014-02-12 15:28:52.000000000 -0600
+++ ff-dev/mozilla-27.0.1/content/base/src/nsFrameLoader.h	2015-01-24 17:44:40.129351476 -0600
@@ -453,6 +453,10 @@
   // See nsIFrameLoader.idl. EVENT_MODE_NORMAL_DISPATCH automatically
   // forwards some input events to out-of-process content.
   uint32_t mEventMode;
+#ifdef TxJS
+public:
+  JS::Value mIntrospector;
+#endif
 };
 
 #endif

