--- ff-orig/mozilla-27.0.1/dom/base/nsIScriptContext.h	2014-02-12 15:28:55.000000000 -0600
+++ ff-dev/mozilla-27.0.1/dom/base/nsIScriptContext.h	2015-01-24 17:44:40.437351467 -0600
@@ -61,12 +61,22 @@
    * @param aOffThreadToken if specified, the result of compiling the script
    *                        on another thread.
    */
+#ifdef TxJS
   virtual nsresult EvaluateString(const nsAString& aScript,
                                   JS::Handle<JSObject*> aScopeObject,
                                   JS::CompileOptions& aOptions,
                                   bool aCoerceToString,
                                   JS::Value* aRetValue,
+                                  const JS::Value &ispect,
                                   void **aOffThreadToken = nullptr) = 0;
+#else
+  virtual nsresult EvaluateString(const nsAString& aScript,
+                                  JS::Handle<JSObject*> aScopeObject,
+                                  JS::CompileOptions& aOptions,
+                                  bool aCoerceToString,
+                                  JS::Value* aRetValue,
+                                  void **aOffThreadToken = nullptr) = 0;
+#endif
 
   /**
    * Bind an already-compiled event handler function to the given

