--- ff-orig/mozilla-27.0.1/js/src/jscntxt.h	2014-02-12 15:29:04.000000000 -0600
+++ ff-dev/mozilla-27.0.1/js/src/jscntxt.h	2015-01-24 17:44:40.861351454 -0600
@@ -432,6 +432,51 @@
     /* Per-context options. */
     JS::ContextOptions  options_;
 
+#ifdef TxJS
+  private:
+    /* Tx data */
+    TxContext *tx_;
+
+    inline TxContext *getTxInState(int state) {
+        TxContext *tx = tx_;
+        while (tx != NULL) {
+            if (tx->inState(state))
+                break;
+            tx = tx->getNext();
+        }
+        return tx;
+    }
+
+  public:
+    inline TxContext *getRunningTx() {
+        return getTxInState(TX_RUNNING);
+    }
+    inline TxContext *getClosingTx() {
+        return getTxInState(TX_RUNNING | TX_DUMMY);
+    }
+    inline TxContext *getSuspendingTx() {
+        return getTxInState(TX_SUSPEND0 | TX_END0);
+    }
+    inline TxContext *getSuspendedTx() {
+        return getTxInState(TX_SUSPEND1 | TX_END1);
+    }
+
+    inline TxContext *getTxContext() { return tx_; }
+    // Return the new transaction to simplify jsinterp.cpp.
+    inline TxContext *setTxContext(TxContext *tx) { tx_ = tx; return tx_; }
+
+    inline const js::Value getIntrospector() {
+        if (currentlyRunningInInterpreter()) {
+            return interpreterFrame()->script()->getIntrospector();
+        }
+        return JS::UndefinedValue();
+    }
+    inline void setIntrospector(const js::Value &ispect) {
+        JS_ASSERT(currentlyRunningInInterpreter());
+        interpreterFrame()->script()->setIntrospector(ispect);
+    }
+#endif
+
   public:
     int32_t             reportGranularity;  /* see vm/Probes.h */
 

