--- ff-orig/mozilla-27.0.1/js/src/jsscript.h	2014-02-12 15:29:04.000000000 -0600
+++ ff-dev/mozilla-27.0.1/js/src/jsscript.h	2015-01-24 17:44:40.941351452 -0600
@@ -64,6 +64,9 @@
     JSTRY_CATCH,
     JSTRY_FINALLY,
     JSTRY_ITER,
+#ifdef TxJS
+    JSTRY_TX,
+#endif
     JSTRY_LOOP
 } JSTryNoteKind;
 
@@ -471,6 +474,16 @@
     /* Persistent type information retained across GCs. */
     js::types::TypeScript *types;
 
+#ifdef TxJS
+  private:
+    js::Value introspector_;
+  public:
+    inline const js::Value getIntrospector() { return introspector_; }
+    inline void setIntrospector(const js::Value &ispect) {
+        introspector_ = ispect;
+    }
+#endif
+
   private:
     js::HeapPtrObject sourceObject_; /* source code object */
     js::HeapPtrFunction function_;

