--- ff-orig/mozilla-27.0.1/js/src/jsatom.h	2014-02-12 15:29:04.000000000 -0600
+++ ff-dev/mozilla-27.0.1/js/src/jsatom.h	2015-01-24 17:44:40.837351455 -0600
@@ -140,6 +140,9 @@
 extern const char js_in_str[];
 extern const char js_instanceof_str[];
 extern const char js_interface_str[];
+#ifdef TxJS
+extern const char js_introspect_str[];
+#endif
 extern const char js_let_str[];
 extern const char js_new_str[];
 extern const char js_package_str[];

