--- ff-orig/mozilla-27.0.1/js/src/vm/Interpreter.h	2014-02-12 15:29:06.000000000 -0600
+++ ff-dev/mozilla-27.0.1/js/src/vm/Interpreter.h	2015-01-24 17:44:41.049351449 -0600
@@ -320,6 +320,20 @@
 extern void
 UnwindScope(JSContext *cx, AbstractFramePtr frame, uint32_t stackDepth);
 
+#ifdef TxJS
+extern bool 
+BeginTx(JSContext *cx, const js::Value &tx_handler);
+
+extern bool 
+IntrospectTx(JSContext *cx, TxContext *tx, InterpreterActivation *activation, bool isEnd, HandleScript script, int *path, bool *newType);
+
+extern bool
+TxSubsumed(JSContext *cx, const js::Value &ispect);
+
+extern bool
+SuspendCall(JSContext *cx, TxContext *tx, CallArgs &args, bool *construct);
+#endif
+
 /*
  * Unwind for an uncatchable exception. This means not running finalizers, etc;
  * just preserving the basic engine stack invariants.

