--- ff-orig/mozilla-27.0.1/js/src/jsopcode.cpp	2014-02-12 15:29:04.000000000 -0600
+++ ff-dev/mozilla-27.0.1/js/src/jsopcode.cpp	2015-01-24 17:44:40.917351453 -0600
@@ -153,8 +153,17 @@
         return NumBlockSlots(script, pc) + 2;
       default:
         /* stack: fun, this, [argc arguments] */
+#ifdef TxJS
+        JS_ASSERT(op == JSOP_NEW || op == JSOP_CALL || op == JSOP_EVAL ||
+                  op == JSOP_BEGIN_TX || op == JSOP_FUNCALL || op == JSOP_FUNAPPLY);
+        // There's no fun or this for transaction entry.
+        if (op == JSOP_BEGIN_TX)
+            return GET_ARGC(pc);
+        else
+#else
         JS_ASSERT(op == JSOP_NEW || op == JSOP_CALL || op == JSOP_EVAL ||
                   op == JSOP_FUNCALL || op == JSOP_FUNAPPLY);
+#endif
         return 2 + GET_ARGC(pc);
     }
 }

