--- ff-orig/mozilla-27.0.1/parser/html/nsHtml5StreamParser.cpp	2014-02-12 15:29:18.000000000 -0600
+++ ff-dev/mozilla-27.0.1/parser/html/nsHtml5StreamParser.cpp	2015-01-24 17:44:41.129351446 -0600
@@ -170,6 +170,9 @@
   , mLoadFlusher(new nsHtml5LoadFlusher(aExecutor))
   , mFlushTimer(do_CreateInstance("@mozilla.org/timer;1"))
   , mMode(aMode)
+#ifdef TxJS
+  , mIntrospector(JS::UndefinedValue())
+#endif
 {
   NS_ASSERTION(NS_IsMainThread(), "Wrong thread!");
   mFlushTimer->SetTarget(mThread);
@@ -894,6 +897,9 @@
   mTreeBuilder->SetPreventScriptExecution(!((mMode == NORMAL) &&
                                             scriptingEnabled));
   mTokenizer->start();
+#ifdef TxJS
+  mExecutor->mIntrospector = mIntrospector;
+#endif
   mExecutor->Start();
   mExecutor->StartReadingFromStage();
 

