--- mozilla-esr17-orig/js/src/jsinterp.h	2013-02-15 14:59:22.000000000 -0600
+++ mozilla-esr17/js/src/jsinterp.h	2014-05-14 19:18:15.448963915 -0500
@@ -207,6 +207,14 @@
 extern JSBool
 HasInstance(JSContext *cx, HandleObject obj, const js::Value *v, JSBool *bp);
 
+#ifdef TxJS
+extern bool 
+BeginTx(JSContext *cx, const js::Value &tx_handler);
+
+extern bool 
+IntrospectTx(JSContext *cx, TxContext *tx, bool isEnd, HandleScript script, int *path, bool *newType);
+#endif
+
 /*
  * A linked list of the |FrameRegs regs;| variables belonging to all
  * js::Interpret C++ frames on this thread's stack.
