--- mozilla-esr17-orig/js/src/jsproto.tbl	2013-02-15 14:59:22.000000000 -0600
+++ mozilla-esr17/js/src/jsproto.tbl	2014-05-02 14:57:37.618561653 -0500
@@ -65,6 +65,10 @@
 JS_PROTO(Set,                   38,     js_InitSetClass)
 JS_PROTO(DataView,              39,     js_InitTypedArrayClasses)
 JS_PROTO(ParallelArray,         40,     js_InitParallelArrayClass)
+#ifdef TxJS
+JS_PROTO(Tx,                    41,     js_InitTxClass)
+JS_PROTO(JAM,                   42,     js_InitJAMClass)
+#endif
 
 #undef XML_INIT
 #undef NAMESPACE_INIT
