--- mozilla-esr17-orig/js/src/frontend/Parser.h	2013-02-15 14:59:19.000000000 -0600
+++ mozilla-esr17/js/src/frontend/Parser.h	2014-05-02 14:57:37.602561653 -0500
@@ -395,6 +395,12 @@
     ParseNode *expr();
     ParseNode *assignExpr();
     ParseNode *assignExprWithoutYield(unsigned err);
+#ifdef TxJS
+    ParseNode *txStmt();
+#ifndef TxJS_NOEXPR
+    ParseNode *txExpr();
+#endif
+#endif
     ParseNode *condExpr1();
     ParseNode *orExpr1();
     ParseNode *andExpr1i();
@@ -427,6 +433,9 @@
     bool functionArguments(ParseNode **list, ParseNode *funcpn, bool &hasRest);
 
     ParseNode *functionDef(HandlePropertyName name, FunctionType type, FunctionSyntaxKind kind);
+#ifdef TxJS
+    bool txArguments(ParseNode *txpn);
+#endif
 
     ParseNode *unaryOpExpr(ParseNodeKind kind, JSOp op);
 
