--- mozilla-esr17-orig/js/src/frontend/SharedContext.h	2013-02-15 14:59:19.000000000 -0600
+++ mozilla-esr17/js/src/frontend/SharedContext.h	2014-05-02 14:57:37.602561653 -0500
@@ -14,6 +14,9 @@
 #include "jsscript.h"
 #include "jsprvtd.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "frontend/ParseMaps.h"
 #include "frontend/ParseNode.h"
@@ -213,6 +216,9 @@
     STMT_IF,                    /* if (then) statement */
     STMT_ELSE,                  /* else clause of if statement */
     STMT_SEQ,                   /* synthetic sequence of statements */
+#ifdef TxJS
+    STMT_TX,                    /* transaction block */
+#endif
     STMT_BLOCK,                 /* compound statement: { s1[;... sN] } */
     STMT_SWITCH,                /* switch statement */
     STMT_WITH,                  /* with statement */
