--- mozilla-esr17-orig/js/src/frontend/TokenStream.h	2013-02-15 14:59:19.000000000 -0600
+++ mozilla-esr17/js/src/frontend/TokenStream.h	2014-05-02 14:57:37.602561653 -0500
@@ -18,6 +18,9 @@
 #include "jsopcode.h"
 #include "jsprvtd.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "js/Vector.h"
 
@@ -156,7 +159,9 @@
     TOK_DIVASSIGN,
     TOK_MODASSIGN,
     TOK_ASSIGNMENT_LAST = TOK_MODASSIGN,
-
+#ifdef TxJS
+    TOK_TX,
+#endif
     TOK_LIMIT                      /* domain size */
 };
 
