--- mozilla-esr17-orig/js/src/jsscript.cpp	2013-02-15 14:59:22.000000000 -0600
+++ mozilla-esr17/js/src/jsscript.cpp	2014-05-23 14:23:41.192758307 -0500
@@ -1508,6 +1508,10 @@
     script->sourceStart = bufStart;
     script->sourceEnd = bufEnd;
 
+#ifdef TxJS
+    script->setIntrospector(UndefinedValue());
+#endif
+
     return script;
 }
 
@@ -2168,6 +2172,9 @@
     dst->isGenerator = src->isGenerator;
     dst->isGeneratorExp = src->isGeneratorExp;
     dst->userBit = src->userBit;
+#ifdef TxJS
+    dst->setIntrospector(src->getIntrospector());
+#endif
 
     /*
      * initScriptCounts updates scriptCountsMap if necessary. The other script
