--- mozilla-esr17-orig/mozconfig-nojit-debug	1969-12-31 18:00:00.000000000 -0600
+++ mozilla-esr17/mozconfig-nojit-debug	2014-05-02 14:57:37.626561652 -0500
@@ -0,0 +1,17 @@
+# Use pound sign for comments.
+ac_add_options --enable-application=browser
+ac_add_options --enable-debug
+ac_add_options --disable-optimize
+#ac_add_options --enable-optimize="-O3 -fno-omit-frame-pointer"
+#ac_add_options --enable-optimize="-O3"
+ac_add_options --disable-methodjit
+ac_add_options --disable-methodjit-spew
+ac_add_options --disable-tracejit
+ac_add_options --disable-jit
+#ac_add_options --enable-vtune
+#ac_add_options --enable-jprof
+#mk_add_options -DLDFLAGS=-lprofiler
+#mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-@CONFIG_GUESS@
+mk_add_options MOZ_OBJDIR=obj-nojit-debug
+mk_add_options MOZ_BUILD_PROJECTS="browser"
+mk_add_options MOZ_MAKE_FLAGS="-j4"
