--- ff-orig/mozilla-24.0esr/js/src/jscntxtinlines.h	2013-09-10 22:22:33.000000000 -0500
+++ ff-dev/mozilla-24.0esr/js/src/jscntxtinlines.h	2015-01-21 18:31:26.733381110 -0600
@@ -317,10 +317,14 @@
 #ifdef DEBUG
     bool alreadyThrowing = cx->isExceptionPending();
 #endif
+#ifndef TxJS
     assertSameCompartment(cx, args);
+#endif
     bool ok = native(cx, args.length(), args.base());
     if (ok) {
+//#ifndef TxJS
         assertSameCompartment(cx, args.rval());
+//#endif
         JS_ASSERT_IF(!alreadyThrowing, !cx->isExceptionPending());
     }
     return ok;

