--- ff-orig/mozilla-24.0esr/js/src/frontend/TokenStream.cpp	2013-09-10 22:22:31.000000000 -0500
+++ ff-dev/mozilla-24.0esr/js/src/frontend/TokenStream.cpp	2015-01-21 18:29:59.537383659 -0600
@@ -1891,6 +1891,9 @@
       case TOK_MULASSIGN:       return "TOK_MULASSIGN";
       case TOK_DIVASSIGN:       return "TOK_DIVASSIGN";
       case TOK_MODASSIGN:       return "TOK_MODASSIGN";
+#ifdef TxJS
+      case TOK_TX:              return "TOK_TX";
+#endif
       case TOK_EXPORT:          return "TOK_EXPORT";
       case TOK_IMPORT:          return "TOK_IMPORT";
       case TOK_LIMIT:           break;

