--- ff-orig/mozilla-24.0esr/js/src/shell/js.cpp	2013-09-10 22:22:33.000000000 -0500
+++ ff-dev/mozilla-24.0esr/js/src/shell/js.cpp	2015-01-21 18:32:16.369379658 -0600
@@ -41,6 +41,10 @@
 #include "jswrapper.h"
 #include "jsperf.h"
 
+// TxJS begin
+#include "jsversion.h"
+// TxJS end
+
 #include "builtin/TestingFunctions.h"
 #include "frontend/BytecodeEmitter.h"
 #include "frontend/Parser.h"
@@ -1841,7 +1845,12 @@
 JS_STATIC_ASSERT(JSTRY_FINALLY == 1);
 JS_STATIC_ASSERT(JSTRY_ITER == 2);
 
+#ifdef TxJS
+JS_STATIC_ASSERT(JSTRY_TX == 3);
+static const char* const TryNoteNames[] = { "catch", "finally", "iter", "tx", "loop" };
+#else
 static const char* const TryNoteNames[] = { "catch", "finally", "iter", "loop" };
+#endif
 
 static JSBool
 TryNotes(JSContext *cx, HandleScript script, Sprinter *sp)
@@ -3607,7 +3616,6 @@
     JS_FN_HELP("readline", ReadLine, 0, 0,
 "readline()",
 "  Read a single line from stdin."),
-
     JS_FN_HELP("print", Print, 0, 0,
 "print([exp ...])",
 "  Evaluate and print expressions to stdout."),

