--- ff-orig/mozilla-17.0.11esr/js/src/jsobjinlines.h	2013-11-13 17:35:25.000000000 -0600
+++ ff-dev/mozilla-17.0.11esr/js/src/jsobjinlines.h	2015-01-18 22:03:23.937204327 -0600
@@ -458,7 +458,18 @@
     JS_ASSERT(dstStart + count <= getDenseArrayCapacity());
     JSCompartment *comp = compartment();
     for (unsigned i = 0; i < count; ++i)
+#ifdef TxJS
+    {
+#endif
         elements[dstStart + i].init(comp, this, dstStart + i, src[i]);
+#ifdef TxJS
+        // %%% To maintain efficiency of dense arrays during
+        // %%% transactions, we might want to create a new ptype for
+        // %%% write sets, and record the speculative writes here.
+        // Callers are responsible for adding the source elements to the
+        // read set.
+    }
+#endif
 }
 
 inline void

