--- ff-orig/mozilla-17.0.11esr/js/src/frontend/Parser-inl.h	2013-11-13 17:35:23.000000000 -0600
+++ ff-dev/mozilla-17.0.11esr/js/src/frontend/Parser-inl.h	2015-01-18 22:03:23.929204327 -0600
@@ -32,6 +32,10 @@
     blockidGen(bodyid),  // used to set |bodyid| and subsequently incremented in init()
     topStmt(NULL),
     topScopeStmt(NULL),
+#ifdef TxJS_HOIST
+    txTop(NULL),
+    hasTxTop(false),
+#endif
     blockChain(prs->context),
     staticLevel(staticLevel),
     parenDepth(0),

