--- ff-orig/mozilla-20.0.1/content/base/src/nsScriptLoader.h	2013-04-09 22:01:23.000000000 -0500
+++ ff-dev/mozilla-20.0.1/content/base/src/nsScriptLoader.h	2015-01-19 21:29:36.682737411 -0600
@@ -18,6 +18,9 @@
 #include "nsAutoPtr.h"
 #include "nsIDocument.h"
 #include "nsIStreamLoader.h"
+// TxJS begin
+#include "jsapi.h"
+// TxJS end
 
 class nsScriptLoadRequest;
 
@@ -207,6 +210,9 @@
                           const nsAString &aCrossOrigin,
                           bool aScriptFromHead);
 
+#ifdef TxJS
+  JS::Value mIntrospector;
+#endif
 private:
   /**
    * Unblocks the creator parser of the parser-blocking scripts.

