--- ff-orig/mozilla-20.0.1/dom/base/nsJSEnvironment.cpp	2013-04-09 22:01:30.000000000 -0500
+++ ff-dev/mozilla-20.0.1/dom/base/nsJSEnvironment.cpp	2015-01-19 21:29:36.758737409 -0600
@@ -1258,6 +1258,25 @@
                                      JS::Value* aRetValue,
                                      bool* aIsUndefined)
 {
+#ifdef TxJS
+  return EvaluateString(aScript, aScopeObject, aPrincipal,
+      aOriginPrincipal, aURL, aLineNo, aVersion, aRetValue,
+      aIsUndefined, JS::UndefinedValue());
+}
+
+nsresult
+nsJSContext::EvaluateString(const nsAString& aScript,
+                            JSObject* aScopeObject,
+                            nsIPrincipal *aPrincipal,
+                            nsIPrincipal *aOriginPrincipal,
+                            const char *aURL,
+                            uint32_t aLineNo,
+                            uint32_t aVersion,
+                            JS::Value* aRetValue,
+                            bool* aIsUndefined,
+                            const JS::Value &ispect)
+{
+#endif
   SAMPLE_LABEL("JS", "EvaluateStringWithValue");
   NS_ABORT_IF_FALSE(aScopeObject,
     "Shouldn't call EvaluateStringWithValue with null scope object.");
@@ -1525,6 +1544,9 @@
            .setPrincipals(nsJSPrincipals::get(principal))
            .setOriginPrincipals(nsJSPrincipals::get(aOriginPrincipal))
            .setVersion(JSVersion(aVersion));
+#ifdef TxJS
+    options.introspector = ispect;
+#endif
     ok = JS::Evaluate(mContext, rootedScope, options,
                       PromiseFlatString(aScript).get(),
                       aScript.Length(), vp);

