--- ff-orig/mozilla-20.0.1/js/src/frontend/TokenStream.h	2013-04-09 22:01:49.000000000 -0500
+++ ff-dev/mozilla-20.0.1/js/src/frontend/TokenStream.h	2015-01-19 21:29:36.826737407 -0600
@@ -21,6 +21,9 @@
 #include "jsopcode.h"
 #include "jsprvtd.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "js/Vector.h"
 
@@ -156,7 +159,9 @@
     TOK_DIVASSIGN,
     TOK_MODASSIGN,
     TOK_ASSIGNMENT_LAST = TOK_MODASSIGN,
-
+#ifdef TxJS
+    TOK_TX,
+#endif
     TOK_LIMIT                      /* domain size */
 };
 

