--- ff-orig/mozilla-20.0.1/js/src/jsarray.h	2013-04-09 22:01:51.000000000 -0500
+++ ff-dev/mozilla-20.0.1/js/src/jsarray.h	2015-01-19 21:29:36.862737406 -0600
@@ -14,6 +14,9 @@
 #include "jspubtd.h"
 #include "jsatom.h"
 #include "jsobj.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 /* Small arrays are dense, no matter what. */
 const unsigned MIN_SPARSE_INDEX = 512;
@@ -156,4 +159,11 @@
 JSBool
 js_Array(JSContext *cx, unsigned argc, js::Value *vp);
 
+#ifdef TxJS
+extern JSBool
+js_SetArrayElement(JSContext *cx, js::HandleObject obj, double index, js::HandleValue vp);
+
+extern JSBool
+js_GetArrayElement(JSContext *cx, js::HandleObject obj, double index, JSBool *hole, js::MutableHandleValue vp);
+#endif
 #endif /* jsarray_h___ */

