--- ff-orig/mozilla-20.0.1/js/src/jsscript.cpp	2013-04-09 22:01:52.000000000 -0500
+++ ff-dev/mozilla-20.0.1/js/src/jsscript.cpp	2015-01-19 21:39:52.110719417 -0600
@@ -1623,6 +1623,10 @@
     script->sourceEnd = bufEnd;
     script->userBit = options.userBit;
 
+#ifdef TxJS
+    script->setIntrospector(UndefinedValue());
+#endif
+
     return script;
 }
 
@@ -2318,6 +2322,9 @@
     dst->hasSingletons = src->hasSingletons;
     dst->isGenerator = src->isGenerator;
     dst->isGeneratorExp = src->isGeneratorExp;
+#ifdef TxJS
+    dst->setIntrospector(src->getIntrospector());
+#endif
 
     /*
      * initScriptCounts updates scriptCountsMap if necessary. The other script

