--- ff-orig/mozilla-20.0.1/js/src/gdb/gdb-tests-gdb.py	1969-12-31 18:00:00.000000000 -0600
+++ ff-dev/mozilla-20.0.1/js/src/gdb/gdb-tests-gdb.py	2015-01-19 22:08:13.742669664 -0600
@@ -0,0 +1,7 @@
+"""GDB Python customization auto-loader for GDB test executable"""
+
+import os.path
+sys.path[0:0] = [os.path.join('/home/rich/cs/jam/jamscript/ff-dev/mozilla-20.0.1/js/src', 'gdb')]
+
+import mozilla.autoload
+mozilla.autoload.register(gdb.current_objfile())

