--- ff-orig/mozilla-20.0.1/js/src/frontend/SharedContext.h	2013-04-09 22:01:49.000000000 -0500
+++ ff-dev/mozilla-20.0.1/js/src/frontend/SharedContext.h	2015-01-19 21:29:36.826737407 -0600
@@ -14,6 +14,9 @@
 #include "jsscript.h"
 #include "jsprvtd.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "frontend/ParseMaps.h"
 #include "frontend/ParseNode.h"
@@ -219,6 +222,9 @@
     STMT_IF,                    /* if (then) statement */
     STMT_ELSE,                  /* else clause of if statement */
     STMT_SEQ,                   /* synthetic sequence of statements */
+#ifdef TxJS
+    STMT_TX,                    /* transaction block */
+#endif
     STMT_BLOCK,                 /* compound statement: { s1[;... sN] } */
     STMT_SWITCH,                /* switch statement */
     STMT_WITH,                  /* with statement */

