--- ff-orig/mozilla-20.0.1/js/src/jsatom.cpp	2013-04-09 22:01:51.000000000 -0500
+++ ff-dev/mozilla-20.0.1/js/src/jsatom.cpp	2015-01-19 21:29:36.866737406 -0600
@@ -112,6 +112,9 @@
 const char js_close_str[]           = "close";
 const char js_send_str[]            = "send";
 #endif
+#ifdef TxJS
+const char js_introspect_str[]      = "introspect";
+#endif
 
 /*
  * For a browser build from 2007-08-09 after the browser starts up there are

