--- ff-orig/mozilla-20.0.1/js/src/frontend/Parser.h	2013-04-09 22:01:49.000000000 -0500
+++ ff-dev/mozilla-20.0.1/js/src/frontend/Parser.h	2015-01-19 21:31:06.898734773 -0600
@@ -17,6 +17,9 @@
 #include "jsatom.h"
 #include "jsscript.h"
 #include "jswin.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "frontend/ParseMaps.h"
 #include "frontend/ParseNode.h"
@@ -59,6 +62,10 @@
 
     StmtInfoPC      *topStmt;       /* top of statement info stack */
     StmtInfoPC      *topScopeStmt;  /* top lexical scope statement */
+#ifdef TxJS_HOIST
+    StmtInfoPC      *txTop;         /* top statement to use for tx */
+    bool             hasTxTop;
+#endif
     Rooted<StaticBlockObject *> blockChain;
                                     /* compile time block scope chain */
 
@@ -395,6 +402,10 @@
     ParseNode *expr();
     ParseNode *assignExpr();
     ParseNode *assignExprWithoutYield(unsigned err);
+#ifdef TxJS
+    ParseNode *txStmt();
+    ParseNode *txExpr();
+#endif
     ParseNode *condExpr1();
     ParseNode *orExpr1();
     ParseNode *andExpr1i();
@@ -428,6 +439,9 @@
 
     ParseNode *functionDef(HandlePropertyName name, const TokenStream::Position &start,
                            FunctionType type, FunctionSyntaxKind kind);
+#ifdef TxJS
+    bool txArguments(ParseNode *txpn);
+#endif
     bool functionArgsAndBody(ParseNode *pn, HandleFunction fun, HandlePropertyName funName,
                              FunctionType type, FunctionSyntaxKind kind, bool strict,
                              bool *becameStrict = NULL);

