--- ff-orig/mozilla-20.0.1/parser/html/nsHtml5StreamParser.cpp	2013-04-09 22:02:41.000000000 -0500
+++ ff-dev/mozilla-20.0.1/parser/html/nsHtml5StreamParser.cpp	2015-01-19 21:29:37.118737398 -0600
@@ -166,6 +166,9 @@
   , mLoadFlusher(new nsHtml5LoadFlusher(aExecutor))
   , mFlushTimer(do_CreateInstance("@mozilla.org/timer;1"))
   , mMode(aMode)
+#ifdef TxJS
+  , mIntrospector(JS::UndefinedValue())
+#endif
 {
   NS_ASSERTION(NS_IsMainThread(), "Wrong thread!");
   mFlushTimer->SetTarget(mThread);
@@ -878,6 +881,9 @@
   mTreeBuilder->SetPreventScriptExecution(!((mMode == NORMAL) &&
                                             scriptingEnabled));
   mTokenizer->start();
+#ifdef TxJS
+  mExecutor->mIntrospector = mIntrospector;
+#endif
   mExecutor->Start();
   mExecutor->StartReadingFromStage();
 

