--- ff-orig/mozilla-31.4.0esr/content/base/src/nsObjectLoadingContent.h	2015-01-05 23:07:58.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/content/base/src/nsObjectLoadingContent.h	2015-02-28 16:28:49.051750808 -0600
@@ -539,7 +539,9 @@
     // as a root for other resource requests.
     nsCOMPtr<nsIURI>            mBaseURI;
 
-
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
 
     // Type of the currently-loaded content.
     ObjectType                  mType           : 8;

