--- ff-orig/mozilla-31.4.0esr/content/html/content/src/nsGenericHTMLFrameElement.cpp	2015-01-05 23:08:00.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/content/html/content/src/nsGenericHTMLFrameElement.cpp	2015-02-01 17:05:48.711502435 -0600
@@ -23,6 +23,10 @@
 #include "nsIScrollable.h"
 #include "nsPresContext.h"
 #include "nsServiceManagerUtils.h"
+#ifdef TxJS
+#include "nsIScriptContext.h"
+#include "jsapi.h"
+#endif
 
 using namespace mozilla;
 using namespace mozilla::dom;
@@ -179,6 +183,16 @@
   }
 
   nsresult rv = mFrameLoader->LoadFrame();
+#ifdef TxJS
+  nsIScriptContext* sctx = GetContextForEventHandlers(&rv);
+  if (sctx) {
+    JSContext* cx = sctx->GetNativeContext();
+    JS::Value ispect = JS_GetIntrospector(cx);
+    if (!ispect.isUndefined()) {
+      mFrameLoader->mIntrospector = ispect;
+    }
+  }
+#endif
 #ifdef DEBUG
   if (NS_FAILED(rv)) {
     NS_WARNING("failed to load URL");

