--- ff-orig/mozilla-31.4.0esr/content/html/content/src/nsGenericHTMLFrameElement.cpp	2015-01-05 23:08:00.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/content/html/content/src/nsGenericHTMLFrameElement.cpp	2015-03-01 11:13:20.478992485 -0600
@@ -23,6 +23,9 @@
 #include "nsIScrollable.h"
 #include "nsPresContext.h"
 #include "nsServiceManagerUtils.h"
+#ifdef TxJS
+#include "jsapi.h"
+#endif
 
 using namespace mozilla;
 using namespace mozilla::dom;
@@ -179,6 +182,13 @@
   }
 
   nsresult rv = mFrameLoader->LoadFrame();
+#ifdef TxJS
+  nsIDocument *doc = GetContentDocument();
+  if (doc) {
+    JSContext *cx = nsContentUtils::GetContextFromDocument(doc);
+    mFrameLoader->mIntrospector = JS_GetIntrospector(cx);
+  }
+#endif
 #ifdef DEBUG
   if (NS_FAILED(rv)) {
     NS_WARNING("failed to load URL");

