--- ff-orig/mozilla-31.4.0esr/content/base/src/nsScriptLoader.h	2015-01-05 23:07:58.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/content/base/src/nsScriptLoader.h	2015-03-01 11:06:23.082997177 -0600
@@ -336,6 +336,11 @@
   uint32_t mBlockerCount;
   bool mEnabled;
   bool mDeferEnabled;
+#ifdef TxJS
+public:
+  JS::Value mIntrospector;
+private:
+#endif
   bool mDocumentParsingDone;
   bool mBlockingDOMContentLoaded;
 };

