--- ff-orig/mozilla-31.4.0esr/content/base/src/nsObjectLoadingContent.h	2015-01-05 23:07:58.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/content/base/src/nsObjectLoadingContent.h	2015-03-01 10:47:45.211009741 -0600
@@ -539,13 +539,15 @@
     // as a root for other resource requests.
     nsCOMPtr<nsIURI>            mBaseURI;
 
-
-
     // Type of the currently-loaded content.
     ObjectType                  mType           : 8;
     // The type of fallback content we're showing (see ObjectState())
     FallbackType                mFallbackType : 8;
 
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
+
     // If true, we have opened a channel as the listener and it has reached
     // OnStartRequest. Does not get set for channels that are passed directly to
     // the plugin listener.

