--- ff-orig/mozilla-31.4.0esr/content/base/src/nsScriptLoader.cpp	2015-01-05 23:07:58.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/content/base/src/nsScriptLoader.cpp	2015-03-01 10:44:36.231011865 -0600
@@ -80,6 +80,9 @@
       mScriptTextLength(0),
       mJSVersion(aVersion),
       mLineNo(1),
+#ifdef TxJS
+      mIntrospector(JS::UndefinedValue()),
+#endif
       mCORSMode(aCORSMode)
   {
   }
@@ -119,6 +122,9 @@
   nsCOMPtr<nsIPrincipal> mOriginPrincipal;
   nsAutoCString mURL;   // Keep the URI's filename alive during off thread parsing.
   int32_t mLineNo;
+#ifdef TxJS
+  JS::Value mIntrospector;
+#endif
   const CORSMode mCORSMode;
 };
 
@@ -136,6 +142,9 @@
     mBlockerCount(0),
     mEnabled(true),
     mDeferEnabled(false),
+#ifdef TxJS
+    mIntrospector(JS::UndefinedValue()),
+#endif
     mDocumentParsingDone(false),
     mBlockingDOMContentLoaded(false)
 {
@@ -643,6 +652,10 @@
     request->mJSVersion = version;
 
     if (aElement->GetScriptAsync()) {
+#ifdef TxJS
+      JSContext *cx = nsContentUtils::GetContextFromDocument(mDocument);
+      request->mIntrospector = JS_GetIntrospector(cx);
+#endif
       mAsyncRequests.AppendElement(request);
       if (!request->mLoading) {
         // The script is available already. Run it ASAP when the event
@@ -738,6 +751,10 @@
   request->mURI = mDocument->GetDocumentURI();
   request->mLineNo = aElement->GetScriptLineNumber();
 
+#ifdef TxJS
+  request->mIntrospector = mIntrospector;
+#endif
+
   if (aElement->GetParserCreated() == FROM_PARSER_XSLT &&
       (!ReadyToExecuteScripts() || !mXSLTRequests.IsEmpty())) {
     // Need to maintain order for XSLT-inserted scripts
@@ -1067,6 +1084,20 @@
     MOZ_ASSERT(elementVal.isObject());
     aOptions->setElement(&elementVal.toObject());
   }
+#ifdef TxJS
+  JS::Value ispect = aRequest->mIntrospector;
+  if (ispect.isUndefined()) {
+    nsAutoString ispectName;
+    mDocument->GetHeaderData(nsGkAtoms::jamIntrospector, ispectName);
+    if (!ispectName.IsEmpty()) {
+      nsIGlobalObject *global = mDocument->GetScopeObject(); 
+      if (global) {
+        fprintf(stderr, "GOT THIS FAR!!\n"/*: %s\n", (char *)ispectName.mStringBuffer->Data()*/);
+      }
+    }
+  }
+  aOptions->setIntrospector(ispect);
+#endif
 }
 
 nsresult

