--- ff-orig/mozilla-31.4.0esr/docshell/base/nsDocShell.cpp	2015-01-05 23:08:02.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/docshell/base/nsDocShell.cpp	2015-03-01 11:21:05.266987262 -0600
@@ -797,6 +797,9 @@
     mDefaultScrollbarPref(Scrollbar_Auto, Scrollbar_Auto),
     mTreeOwner(nullptr),
     mChromeEventHandler(nullptr),
+#ifdef TxJS
+    mIntrospector(JS::UndefinedValue()),
+#endif
     mCharsetReloadState(eCharsetReloadInit),
     mChildOffset(0),
     mBusyFlags(BUSY_FLAGS_NONE),
@@ -1798,6 +1801,26 @@
         originIsFile && targetIsFile;
 }
 
+// TxJS begin
+NS_IMETHODIMP
+nsDocShell::GetIntrospector(JS::MutableHandleValue aIntrospector) {
+#ifdef TxJS
+    aIntrospector.set(mIntrospector);
+#else
+    aIntrospector.set(JS::UndefinedValue());
+#endif
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsDocShell::SetIntrospector(JS::HandleValue aIntrospector) {
+#ifdef TxJS
+    mIntrospector = aIntrospector;
+#endif
+    return NS_OK;
+}
+// TxJS end
+
 NS_IMETHODIMP
 nsDocShell::GetEldestPresContext(nsPresContext** aPresContext)
 {
@@ -10031,6 +10054,18 @@
             SetExecutionPolicy(nsIScriptChannel::EXECUTE_NORMAL);
     }
 
+#ifdef TxJS
+    JSContext *cx = nsContentUtils::GetCurrentJSContext();
+    if (cx) {
+        JS::RootedValue ispect(cx);
+        GetIntrospector(&ispect);
+        // %%% Not sure if we need to do both.
+        if (scriptChannel)
+            channel->SetIntrospector(ispect);
+        uriLoader->SetIntrospector(ispect);
+    }
+#endif
+
     if (aIsNewWindowTarget) {
         nsCOMPtr<nsIWritablePropertyBag2> props = do_QueryInterface(channel);
         if (props) {
@@ -10047,7 +10082,6 @@
             timedChannel->SetInitiatorType(NS_LITERAL_STRING("subdocument"));
         }
     }
-
     rv = DoChannelLoad(channel, uriLoader, aBypassClassifier);
 
     //

