--- ff-orig/mozilla-31.4.0esr/dom/src/jsurl/nsJSProtocolHandler.cpp	2015-01-05 23:08:05.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/dom/src/jsurl/nsJSProtocolHandler.cpp	2015-02-01 14:31:10.023813188 -0600
@@ -65,6 +65,9 @@
                             PopupControlState aPopupState,
                             uint32_t aExecutionPolicy,
                             nsPIDOMWindow *aOriginalInnerWindow);
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
 
 protected:
     virtual ~nsJSThunk();
@@ -100,6 +103,10 @@
     rv = uri->GetSpec(mURL);
     if (NS_FAILED(rv)) return rv;
 
+#ifdef TxJS
+    mIntrospector = JS::UndefinedValue();
+#endif
+
     return NS_OK;
 }
 
@@ -314,6 +321,9 @@
                .setVersion(JSVERSION_DEFAULT);
         nsJSUtils::EvaluateOptions evalOptions;
         evalOptions.setCoerceToString(true);
+#ifdef TxJS
+        options.setIntrospector(mIntrospector);
+#endif
         rv = nsJSUtils::EvaluateString(cx, NS_ConvertUTF8toUTF16(script),
                                        globalJSObject, options, evalOptions, &v);
 
@@ -426,6 +436,9 @@
     uint32_t                mExecutionPolicy;
     bool                    mIsAsync;
     bool                    mIsActive;
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
     bool                    mOpenedStreamChannel;
 };
 
@@ -437,6 +450,9 @@
     mExecutionPolicy(EXECUTE_IN_SANDBOX),
     mIsAsync(true),
     mIsActive(false),
+#ifdef TxJS
+    mIntrospector(JS::UndefinedValue()),
+#endif
     mOpenedStreamChannel(false)
 {
 }
@@ -505,6 +521,28 @@
                   nsIStreamListener, nsIScriptChannel, nsIPropertyBag,
                   nsIPropertyBag2)
 
+// TxJS begin
+NS_IMETHODIMP
+nsJSChannel::GetIntrospector(JS::MutableHandleValue aIntrospector)
+{
+#ifdef TxJS
+    aIntrospector.set(mIntrospector);
+#else
+    aIntrospector.set(JS::UndefinedValue());
+#endif
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsJSChannel::SetIntrospector(JS::HandleValue aIntrospector)
+{
+#ifdef TxJS
+    mIntrospector = aIntrospector;
+#endif
+    return NS_OK;
+}
+// TxJS end
+
 //
 // nsIRequest implementation...
 //
@@ -618,6 +656,17 @@
 
     mIsActive = true;
 
+#ifdef TxJS
+    nsCOMPtr<nsIScriptContext> scriptContext = global->GetContext();
+    if (scriptContext) {
+        JSContext* cx = scriptContext->GetNativeContext();
+        JS::Value ispect = JS_GetIntrospector(cx);
+        if (!ispect.isUndefined()) {
+            SetIntrospector(JS::MutableHandleValue::fromMarkedLocation(&ispect));
+        }
+    }
+#endif
+
     // Temporarily set the LOAD_BACKGROUND flag to suppress load group observer
     // notifications (and hence nsIWebProgressListener notifications) from
     // being dispatched.  This is required since we suppress LOAD_DOCUMENT_URI,
@@ -716,6 +765,9 @@
     // script returns it).
     
     if (NS_SUCCEEDED(mStatus)) {
+#ifdef TxJS
+        GetIntrospector(JS::MutableHandleValue::fromMarkedLocation(&mIOThunk->mIntrospector));
+#endif
         nsresult rv = mIOThunk->EvaluateScript(mStreamChannel, mPopupState,
                                                mExecutionPolicy,
                                                mOriginalInnerWindow);

