--- ff-orig/mozilla-31.4.0esr/js/src/jsprototypes.h	2015-01-05 23:08:03.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/js/src/jsprototypes.h	2015-02-02 17:01:22.288617730 -0600
@@ -60,6 +60,12 @@
 #define IF_SAB(real,imaginary) imaginary
 #endif
 
+//#ifdef TxJS
+#define IF_TxJS(real,imaginary) real
+//#else
+//#define IF_TxJS(real,imaginary) imaginary
+//#endif
+
 #define JS_FOR_PROTOTYPES(real,imaginary) \
     imaginary(Null,              0,     js_InitNullClass,          dummy) \
     real(Object,                 1,     js_InitObjectClass,        &JSObject::class_) \
@@ -102,6 +108,8 @@
 IF_BDATA(real,imaginary)(TypedObject,           38,     js_InitTypedObjectModuleObject,   OCLASP(TypedObjectModule)) \
     imaginary(GeneratorFunction,     39,     js_InitIteratorClasses, dummy) \
 IF_BDATA(real,imaginary)(SIMD,                  40,     js_InitSIMDClass, OCLASP(SIMD)) \
+IF_TxJS(real,imaginary)(Tx,                     41,     js_InitTxClass,            CLASP(Tx)) \
+IF_TxJS(real,imaginary)(JAM,                    42,     js_InitJAMClass,           CLASP(JAM)) \
 
 #define JS_FOR_EACH_PROTOTYPE(macro) JS_FOR_PROTOTYPES(macro,macro)
 

