--- ff-orig/mozilla-31.4.0esr/js/src/frontend/TokenStream.cpp	2015-01-05 23:08:00.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/js/src/frontend/TokenStream.cpp	2015-01-28 15:21:05.462440553 -0600
@@ -1811,6 +1811,9 @@
       case TOK_MULASSIGN:       return "TOK_MULASSIGN";
       case TOK_DIVASSIGN:       return "TOK_DIVASSIGN";
       case TOK_MODASSIGN:       return "TOK_MODASSIGN";
+#ifdef TxJS
+      case TOK_TX:              return "TOK_TX";
+#endif
       case TOK_EXPORT:          return "TOK_EXPORT";
       case TOK_IMPORT:          return "TOK_IMPORT";
       case TOK_LIMIT:           break;

