--- ff-orig/mozilla-31.4.0esr/js/src/jsscript.cpp	2015-01-05 23:08:03.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/js/src/jsscript.cpp	2015-02-02 15:06:59.324847578 -0600
@@ -2295,6 +2295,10 @@
     script->sourceStart_ = bufStart;
     script->sourceEnd_ = bufEnd;
 
+#ifdef TxJS
+    script->setIntrospector(UndefinedValue());
+#endif
+
     return script;
 }
 
@@ -3012,6 +3016,9 @@
     dst->hasSingletons_ = src->hasSingletons();
     dst->treatAsRunOnce_ = src->treatAsRunOnce();
     dst->isGeneratorExp_ = src->isGeneratorExp();
+#ifdef TxJS
+    dst->setIntrospector(src->getIntrospector());
+#endif
     dst->setGeneratorKind(src->generatorKind());
 
     /* Copy over hints. */

