--- ff-orig/mozilla-31.4.0esr/js/src/jsversion.h	2015-01-05 23:08:03.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/js/src/jsversion.h	2015-02-01 16:12:46.391609014 -0600
@@ -26,6 +26,9 @@
 #define JS_HAS_GENERATOR_EXPRS  1       /* has (expr for (lhs in iterable)) */
 #define JS_HAS_EXPR_CLOSURES    1       /* has function (formals) listexpr */
 
+/* Enable transactions */
+#define TxJS 1
+
 /* Support for JS_NewGlobalObject. */
 #define JS_HAS_NEW_GLOBAL_OBJECT        1
 

