--- ff-orig/mozilla-31.4.0esr/js/src/jsmath.cpp	2015-01-05 23:08:03.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/js/src/jsmath.cpp	2015-01-27 08:15:59.701091993 -0600
@@ -1441,6 +1441,47 @@
 #endif
 
 static const JSFunctionSpec math_static_methods[] = {
+#ifdef TxJS
+#if JS_HAS_TOSOURCE
+    JS_FN(js_toSource_str,  math_toSource,        0, JSFUN_TX_NOSUSPEND),
+#endif
+    JS_FN("abs",            js_math_abs,          1, JSFUN_TX_NOSUSPEND),
+    JS_FN("acos",           math_acos,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("asin",           math_asin,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("atan",           math_atan,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("atan2",          math_atan2,           2, JSFUN_TX_NOSUSPEND),
+    JS_FN("ceil",           math_ceil,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("cos",            math_cos,             1, JSFUN_TX_NOSUSPEND),
+    JS_FN("exp",            math_exp,             1, JSFUN_TX_NOSUSPEND),
+    JS_FN("floor",          math_floor,           1, JSFUN_TX_NOSUSPEND),
+    JS_FN("imul",           math_imul,            2, JSFUN_TX_NOSUSPEND),
+    JS_FN("log",            math_log,             1, JSFUN_TX_NOSUSPEND),
+    JS_FN("max",            js_math_max,          2, JSFUN_TX_NOSUSPEND),
+    JS_FN("min",            js_math_min,          2, JSFUN_TX_NOSUSPEND),
+    JS_FN("pow",            js_math_pow,          2, JSFUN_TX_NOSUSPEND),
+    JS_FN("random",         js_math_random,       0, 0),
+    JS_FN("round",          math_round,           1, JSFUN_TX_NOSUSPEND),
+    JS_FN("sin",            math_sin,             1, JSFUN_TX_NOSUSPEND),
+    JS_FN("sqrt",           js_math_sqrt,         1, JSFUN_TX_NOSUSPEND),
+    JS_FN("tan",            math_tan,             1, JSFUN_TX_NOSUSPEND),
+    JS_FN("log10",          math_log10,           1, JSFUN_TX_NOSUSPEND),
+    JS_FN("log2",           math_log2,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("log1p",          math_log1p,           1, JSFUN_TX_NOSUSPEND),
+    JS_FN("expm1",          math_expm1,           1, JSFUN_TX_NOSUSPEND),
+    JS_FN("cosh",           math_cosh,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("sinh",           math_sinh,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("tanh",           math_tanh,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("acosh",          math_acosh,           1, JSFUN_TX_NOSUSPEND),
+    JS_FN("asinh",          math_asinh,           1, JSFUN_TX_NOSUSPEND),
+    JS_FN("atanh",          math_atanh,           1, JSFUN_TX_NOSUSPEND),
+// Math.hypot is disabled pending the resolution of spec issues (bug 896264).
+#if 0
+    JS_FN("hypot",          math_hypot,           2, JSFUN_TX_NOSUSPEND),
+#endif
+    JS_FN("trunc",          math_trunc,           1, JSFUN_TX_NOSUSPEND),
+    JS_FN("sign",           math_sign,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("cbrt",           math_cbrt,            1, JSFUN_TX_NOSUSPEND),
+#else
 #if JS_HAS_TOSOURCE
     JS_FN(js_toSource_str,  math_toSource,        0, 0),
 #endif
@@ -1479,6 +1520,7 @@
     JS_FN("trunc",          math_trunc,           1, 0),
     JS_FN("sign",           math_sign,            1, 0),
     JS_FN("cbrt",           math_cbrt,            1, 0),
+#endif
     JS_FS_END
 };
 

