--- ff-orig/mozilla-31.4.0esr/js/src/js/src/js.pc	1969-12-31 18:00:00.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/js/src/js/src/js.pc	2015-01-27 08:15:59.705091993 -0600
@@ -0,0 +1,11 @@
+prefix=/usr/local
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: SpiderMonkey 31.4.0
+Description: The Mozilla library for JavaScript
+Version: 31.4.0
+Requires.private: nspr
+Libs: -L${libdir} -lmozjs-31
+Cflags: -include ${includedir}/mozjs-31/js/RequiredDefines.h -I${includedir}/mozjs-31

