--- ff-orig/mozilla-31.4.0esr/netwerk/streamconv/converters/nsMultiMixedConv.cpp	2015-01-05 23:07:57.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/netwerk/streamconv/converters/nsMultiMixedConv.cpp	2015-02-28 22:19:39.303514221 -0600
@@ -239,6 +239,19 @@
     return NS_OK;
 }
 
+// TxJS begin
+NS_IMETHODIMP
+nsPartChannel::GetIntrospector(JS::MutableHandleValue aIntrospector) {
+    aIntrospector.set(JS::UndefinedValue());
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsPartChannel::SetIntrospector(JS::HandleValue aIntrospector) {
+    return NS_ERROR_NOT_IMPLEMENTED;
+}
+// TxJS end
+
 NS_IMETHODIMP
 nsPartChannel::GetOwner(nsISupports* *aOwner)
 {

