--- ff-orig/mozilla-31.4.0esr/netwerk/base/src/nsAsyncStreamCopier.cpp	2015-01-05 23:07:57.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/netwerk/base/src/nsAsyncStreamCopier.cpp	2015-02-28 22:06:58.547522771 -0600
@@ -228,6 +228,19 @@
     return NS_OK;
 }
 
+// TxJS begin
+NS_IMETHODIMP
+nsAsyncStreamCopier::GetIntrospector(JS::MutableHandleValue aIntrospector) {
+    aIntrospector.set(JS::UndefinedValue());
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsAsyncStreamCopier::SetIntrospector(JS::HandleValue aIntrospector) {
+    return NS_ERROR_NOT_IMPLEMENTED;
+}
+// TxJS end
+
 NS_IMETHODIMP
 nsAsyncStreamCopier::SetLoadGroup(nsILoadGroup *aLoadGroup)
 {

