--- ff-orig/mozilla-31.4.0esr/netwerk/base/src/nsInputStreamPump.cpp	2015-01-05 23:07:57.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/netwerk/base/src/nsInputStreamPump.cpp	2015-02-28 22:11:43.579519568 -0600
@@ -288,6 +288,19 @@
     return NS_OK;
 }
 
+// TxJS begin
+NS_IMETHODIMP
+nsInputStreamPump::GetIntrospector(JS::MutableHandleValue aIntrospector) {
+  aIntrospector.set(JS::UndefinedValue());
+  return NS_OK;
+}
+
+NS_IMETHODIMP
+nsInputStreamPump::SetIntrospector(JS::HandleValue aIntrospector) {
+  return NS_ERROR_NOT_IMPLEMENTED;
+}
+// TxJS end
+
 //-----------------------------------------------------------------------------
 // nsInputStreamPump::nsIInputStreamPump implementation
 //-----------------------------------------------------------------------------

