--- ff-orig/mozilla-31.4.0esr/parser/html/nsHtml5TreeOpExecutor.cpp	2015-01-05 23:07:58.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/parser/html/nsHtml5TreeOpExecutor.cpp	2015-03-01 11:39:04.366975133 -0600
@@ -62,6 +62,9 @@
 
 nsHtml5TreeOpExecutor::nsHtml5TreeOpExecutor()
   : nsHtml5DocumentBuilder(false)
+#ifdef TxJS
+  , mIntrospector(JS::UndefinedValue())
+#endif
   , mPreloadedURLs(23)  // Mean # of preloadable resources per page on dmoz
 {
   // zeroing operator new for everything else
@@ -636,6 +639,13 @@
 
   NS_ASSERTION(aScriptElement, "No script to run");
   nsCOMPtr<nsIScriptElement> sele = do_QueryInterface(aScriptElement);
+#ifdef TxJS
+  if (!mIntrospector.isUndefined()) {
+    nsIDocument* ownerDoc = aScriptElement->OwnerDoc();
+    nsRefPtr<nsScriptLoader> loader = ownerDoc->ScriptLoader();
+    loader->mIntrospector = mIntrospector;
+  }
+#endif
   
   if (!mParser) {
     NS_ASSERTION(sele->IsMalformed(), "Script wasn't marked as malformed.");

