--- ff-orig/mozilla-31.4.0esr/parser/html/nsHtml5StreamParser.h	2015-01-05 23:07:58.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/parser/html/nsHtml5StreamParser.h	2015-02-01 14:28:08.603819264 -0600
@@ -19,6 +19,9 @@
 #include "nsHtml5Speculation.h"
 #include "nsITimer.h"
 #include "nsICharsetDetector.h"
+#if 0//def TxJS
+#include "jsapi.h"
+#endif
 
 class nsHtml5Parser;
 
@@ -550,6 +553,11 @@
      * timer has already fired previously in this parse.
      */
     static int32_t                sTimerSubsequentDelay;
+
+#ifdef TxJS
+  public:
+    JS::Value mIntrospector;
+#endif
 };
 
 #endif // nsHtml5StreamParser_h

