--- ff-orig/mozilla-31.4.0esr/parser/html/nsHtml5TreeOpExecutor.h	2015-01-05 23:07:58.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/parser/html/nsHtml5TreeOpExecutor.h	2015-03-01 11:38:04.558975806 -0600
@@ -41,6 +41,10 @@
     NS_DECL_AND_IMPL_ZEROING_OPERATOR_NEW
     NS_DECL_ISUPPORTS_INHERITED
 
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
+
   private:
     static bool        sExternalViewSource;
 #ifdef DEBUG_NS_HTML5_TREE_OP_EXECUTOR_FLUSH

