--- ff-orig/mozilla-31.4.0esr/parser/html/nsHtml5StreamParser.cpp	2015-01-05 23:07:58.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/parser/html/nsHtml5StreamParser.cpp	2015-03-01 11:35:10.202977765 -0600
@@ -886,6 +886,9 @@
   mTreeBuilder->SetPreventScriptExecution(!((mMode == NORMAL) &&
                                             scriptingEnabled));
   mTokenizer->start();
+#ifdef TxJS
+  aRequest->GetIntrospector(JS::MutableHandleValue::fromMarkedLocation(&mExecutor->mIntrospector));
+#endif
   mExecutor->Start();
   mExecutor->StartReadingFromStage();
 

