--- ff-orig/mozilla-31.4.0esr/uriloader/exthandler/ExternalHelperAppParent.cpp	2015-01-05 23:08:07.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/uriloader/exthandler/ExternalHelperAppParent.cpp	2015-02-28 22:26:55.639509317 -0600
@@ -308,6 +308,19 @@
   return NS_ERROR_NOT_IMPLEMENTED;
 }
 
+// TxJS begin
+NS_IMETHODIMP
+ExternalHelperAppParent::GetIntrospector(JS::MutableHandleValue aIntrospector) {
+    aIntrospector.set(JS::UndefinedValue());
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+ExternalHelperAppParent::SetIntrospector(JS::HandleValue aIntrospector) {
+    return NS_ERROR_NOT_IMPLEMENTED;
+}
+// TxJS end
+
 NS_IMETHODIMP
 ExternalHelperAppParent::GetOwner(nsISupports* *aOwner)
 {

