--- ff-orig/mozilla-31.4.0esr/uriloader/base/nsURILoader.cpp	2015-01-05 23:08:07.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/uriloader/base/nsURILoader.cpp	2015-02-01 16:31:55.955570514 -0600
@@ -48,6 +48,9 @@
 #include "nsMimeTypes.h"
 
 #include "nsDocLoader.h"
+#ifdef TxJS
+#include "nsHtml5StreamParser.h"
+#endif
 #include "mozilla/Attributes.h"
 #include "mozilla/Preferences.h"
 
@@ -159,6 +162,11 @@
    * nsIURIContentListeners.
    */
   nsRefPtr<nsURILoader> mURILoader;
+
+#ifdef TxJS
+public:
+  JS::Value mIntrospector;
+#endif
 };
 
 NS_IMPL_ADDREF(nsDocumentOpenInfo)
@@ -181,6 +189,9 @@
                                        nsURILoader* aURILoader)
   : m_originalContext(aWindowContext),
     mFlags(aFlags),
+#ifdef TxJS
+    mIntrospector(JS::UndefinedValue()),
+#endif
     mURILoader(aURILoader)
 {
 }
@@ -729,6 +740,14 @@
     LOG(("  Listener has aborted the load"));
     m_targetStreamListener = nullptr;
   }
+#ifdef TxJS
+  else if (!mIntrospector.isUndefined()) {
+    if (mContentType.LowerCaseEqualsASCII(TEXT_HTML)) {
+      nsHtml5StreamParser *l = (nsHtml5StreamParser *)m_targetStreamListener.get();
+      l->mIntrospector = mIntrospector;
+    }
+  }
+#endif
 
   NS_ASSERTION(abort || m_targetStreamListener, "DoContent returned no listener?");
 
@@ -748,6 +767,9 @@
     mLog = PR_NewLogModule("URILoader");
   }
 #endif
+#ifdef TxJS
+  mIntrospector = JS::UndefinedValue();
+#endif
 }
 
 nsURILoader::~nsURILoader()
@@ -869,6 +891,9 @@
   // the url and discover the content type....
   nsRefPtr<nsDocumentOpenInfo> loader =
     new nsDocumentOpenInfo(aWindowContext, aFlags, this);
+#ifdef TxJS
+  loader->mIntrospector = mIntrospector;
+#endif
 
   if (!loader) return NS_ERROR_OUT_OF_MEMORY;
 

