--- ff-orig/mozilla-31.4.0esr/uriloader/base/nsURILoader.cpp	2015-01-05 23:08:07.000000000 -0600
+++ ff-dev/mozilla-31.4.0esr/uriloader/base/nsURILoader.cpp	2015-02-28 19:13:54.623639478 -0600
@@ -48,6 +48,9 @@
 #include "nsMimeTypes.h"
 
 #include "nsDocLoader.h"
+#ifdef TxJS
+#include "nsHtml5StreamParser.h"
+#endif
 #include "mozilla/Attributes.h"
 #include "mozilla/Preferences.h"
 
@@ -159,6 +162,11 @@
    * nsIURIContentListeners.
    */
   nsRefPtr<nsURILoader> mURILoader;
+
+#ifdef TxJS
+public:
+  JS::Value mIntrospector;
+#endif
 };
 
 NS_IMPL_ADDREF(nsDocumentOpenInfo)
@@ -182,6 +190,9 @@
   : m_originalContext(aWindowContext),
     mFlags(aFlags),
     mURILoader(aURILoader)
+#ifdef TxJS
+    , mIntrospector(JS::UndefinedValue())
+#endif
 {
 }
 
@@ -268,6 +279,9 @@
 
   NS_ENSURE_SUCCESS(rv, rv);
   
+#ifdef TxJS
+  request->SetIntrospector(JS::HandleValue::fromMarkedLocation(&mIntrospector));
+#endif
   if (m_targetStreamListener)
     rv = m_targetStreamListener->OnStartRequest(request, aCtxt);
 
@@ -748,6 +762,9 @@
     mLog = PR_NewLogModule("URILoader");
   }
 #endif
+#ifdef TxJS
+  mIntrospector = JS::UndefinedValue();
+#endif
 }
 
 nsURILoader::~nsURILoader()
@@ -869,6 +886,9 @@
   // the url and discover the content type....
   nsRefPtr<nsDocumentOpenInfo> loader =
     new nsDocumentOpenInfo(aWindowContext, aFlags, this);
+#ifdef TxJS
+  loader->mIntrospector = mIntrospector;
+#endif
 
   if (!loader) return NS_ERROR_OUT_OF_MEMORY;
 

