--- ff-orig/mozilla-17.0.3esr/docshell/base/nsIDocShell.idl	2013-02-15 14:59:08.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/docshell/base/nsIDocShell.idl	2015-01-18 12:12:50.014240354 -0600
@@ -165,6 +165,10 @@
    */ 
   void setCurrentURI(in nsIURI aURI);
 
+  // TxJS begin
+  attribute jsval introspector;
+  // TxJS end
+
   /**
    * Notify the associated content viewer and all child docshells that they are
    * about to be hidden.  If |isUnload| is true, then the document is being

