--- ff-orig/mozilla-17.0.3esr/docshell/base/nsDocShell.h	2013-02-15 14:59:08.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/docshell/base/nsDocShell.h	2015-01-18 12:12:50.014240354 -0600
@@ -756,6 +756,10 @@
     nsIDocShellTreeOwner *     mTreeOwner; // Weak Reference
     nsIDOMEventTarget *       mChromeEventHandler; //Weak Reference
 
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
+
     eCharsetReloadState        mCharsetReloadState;
 
     // Offset in the parent's child list.

