--- ff-orig/mozilla-17.0.3esr/js/src/jskeyword.tbl	2013-02-15 14:59:22.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/js/src/jskeyword.tbl	2015-01-18 12:12:50.042240353 -0600
@@ -65,6 +65,10 @@
 JS_KEYWORD(yield,       TOK_STRICT_RESERVED, JSOP_NOP,  JSVERSION_1_7)
 #endif
 
+#ifdef TxJS
+JS_KEYWORD(introspect, TOK_TX,              JSOP_NOP,  JSVERSION_DEFAULT)
+#endif
+
 /* ES5 future reserved keywords in strict mode. */
 JS_KEYWORD(implements,  TOK_STRICT_RESERVED, JSOP_NOP,  JSVERSION_DEFAULT)
 JS_KEYWORD(interface,   TOK_STRICT_RESERVED, JSOP_NOP,  JSVERSION_DEFAULT)

