--- ff-orig/mozilla-17.0.3esr/js/src/jscntxt.cpp	2013-02-15 14:59:21.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/js/src/jscntxt.cpp	2015-01-18 12:12:50.026240354 -0600
@@ -1083,6 +1083,9 @@
     throwing(false),
     exception(UndefinedValue()),
     runOptions(0),
+#ifdef TxJS
+    tx_(NULL),
+#endif
     reportGranularity(JS_DEFAULT_JITREPORT_GRANULARITY),
     localeCallbacks(NULL),
     resolvingList(NULL),
@@ -1419,6 +1422,11 @@
     TraceCycleDetectionSet(trc, cycleDetectorSet);
 
     MarkValueRoot(trc, &iterValue, "iterValue");
+#ifdef TxJS
+    if (tx_) {
+        tx_->trace(trc);
+    }
+#endif
 }
 
 namespace JS {

