--- mozilla-esr17-orig/js/src/jscntxtinlines.h	2013-02-15 14:59:21.000000000 -0600
+++ mozilla-esr17/js/src/jscntxtinlines.h	2014-06-25 15:44:00.710776124 -0500
@@ -368,10 +368,14 @@
 #ifdef DEBUG
     bool alreadyThrowing = cx->isExceptionPending();
 #endif
+#ifndef TxJS
     assertSameCompartment(cx, args);
+#endif
     bool ok = native(cx, args.length(), args.base());
     if (ok) {
+//#ifndef TxJS
         assertSameCompartment(cx, args.rval());
+//#endif
         JS_ASSERT_IF(!alreadyThrowing, !cx->isExceptionPending());
     }
     return ok;
