--- ff-orig/mozilla-17.0.3esr/js/src/jsatom.tbl	2013-02-15 14:59:21.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/js/src/jsatom.tbl	2015-01-18 12:12:50.022240354 -0600
@@ -152,3 +152,19 @@
 DEFINE_ATOM(unwatch, "unwatch")
 DEFINE_ATOM(watch, "watch")
 DEFINE_ATOM(_CallFunction, "_CallFunction")
+#ifdef TxJS
+DEFINE_ATOM(type, "type")
+DEFINE_ATOM(obj, "obj")
+DEFINE_ATOM(id, "id")
+DEFINE_ATOM(argc, "argc")
+DEFINE_ATOM(args, "args")
+DEFINE_ATOM(description, "description")
+DEFINE_ATOM(subsumedBy, "subsumedBy")
+DEFINE_ATOM(read, "read")
+DEFINE_ATOM(write, "write")
+DEFINE_ATOM(forin, "forin")
+//#ifdef TxJS_ITYPE
+DEFINE_ATOM(itype, "itype")
+//#endif
+DEFINE_KEYWORD_ATOM(new)
+#endif

