--- ff-orig/mozilla-17.0.3esr/js/src/jsstr.cpp	2013-02-15 14:59:22.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/js/src/jsstr.cpp	2015-01-18 12:12:50.046240353 -0600
@@ -3233,7 +3233,11 @@
 }
 
 static JSFunctionSpec string_static_methods[] = {
+#ifdef TxJS
+    JS_FN("fromCharCode", js::str_fromCharCode, 1, JSFUN_TX_NOSUSPEND),
+#else
     JS_FN("fromCharCode", js::str_fromCharCode, 1, 0),
+#endif
     JS_FS_END
 };
 

