--- ff-orig/mozilla-17.0.3esr/js/src/jsinfer.cpp	2013-02-15 14:59:22.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/js/src/jsinfer.cpp	2015-01-18 12:12:50.042240353 -0600
@@ -3471,6 +3471,10 @@
       case JSOP_LOOKUPSWITCH:
       case JSOP_TRY:
       case JSOP_LABEL:
+#ifdef TxJS
+      case JSOP_BEGIN_TX:
+      case JSOP_END_TX:
+#endif
         break;
 
         /* Bytecodes pushing values of known type. */
@@ -5409,6 +5413,12 @@
 /* static */ void
 TypeScript::CheckBytecode(JSContext *cx, JSScript *script, jsbytecode *pc, const js::Value *sp)
 {
+#ifdef TxJS
+    // The host can potentially push any value as the result of
+    // evaluating a suspending expression.
+    //if (cx->getRunningTx())
+    return;
+#endif
     AutoEnterTypeInference enter(cx);
 
     if (js_CodeSpec[*pc].format & JOF_DECOMPOSE)

