--- ff-orig/mozilla-17.0.3esr/js/src/Makefile.in	2013-02-15 14:59:19.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/js/src/Makefile.in	2015-01-18 12:12:50.050240353 -0600
@@ -85,6 +85,7 @@
 		jsinfer.cpp \
 		jsinterp.cpp \
 		jsiter.cpp \
+    jsjam.cpp \
 		jslog2.cpp \
 		jsmath.cpp \
 		jsnativestack.cpp \
@@ -102,6 +103,9 @@
 		jsscope.cpp \
 		jsscript.cpp \
 		jsstr.cpp \
+    jstransaction.cpp \
+    jstx.cpp \
+    jstxaction.cpp \
 		jstypedarray.cpp \
 		jsutil.cpp \
 		jswatchpoint.cpp \

