--- ff-orig/mozilla-17.0.3esr/js/src/jscntxtinlines.h	2013-02-15 14:59:21.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/js/src/jscntxtinlines.h	2015-01-18 12:12:50.026240354 -0600
@@ -368,10 +368,14 @@
 #ifdef DEBUG
     bool alreadyThrowing = cx->isExceptionPending();
 #endif
+#ifndef TxJS
     assertSameCompartment(cx, args);
+#endif
     bool ok = native(cx, args.length(), args.base());
     if (ok) {
+//#ifndef TxJS
         assertSameCompartment(cx, args.rval());
+//#endif
         JS_ASSERT_IF(!alreadyThrowing, !cx->isExceptionPending());
     }
     return ok;

