--- ff-orig/mozilla-17.0.3esr/js/src/jsscript.cpp	2013-02-15 14:59:22.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/js/src/jsscript.cpp	2015-01-18 12:12:50.046240353 -0600
@@ -1508,6 +1508,10 @@
     script->sourceStart = bufStart;
     script->sourceEnd = bufEnd;
 
+#ifdef TxJS
+    script->setIntrospector(UndefinedValue());
+#endif
+
     return script;
 }
 
@@ -2168,6 +2172,9 @@
     dst->isGenerator = src->isGenerator;
     dst->isGeneratorExp = src->isGeneratorExp;
     dst->userBit = src->userBit;
+#ifdef TxJS
+    dst->setIntrospector(src->getIntrospector());
+#endif
 
     /*
      * initScriptCounts updates scriptCountsMap if necessary. The other script

