--- ff-orig/mozilla-17.0.3esr/js/src/frontend/Parser-inl.h	2013-02-15 14:59:19.000000000 -0600
+++ ff-dev/mozilla-17.0.3esr/js/src/frontend/Parser-inl.h	2015-01-18 17:54:14.965641409 -0600
@@ -32,6 +32,10 @@
     blockidGen(bodyid),  // used to set |bodyid| and subsequently incremented in init()
     topStmt(NULL),
     topScopeStmt(NULL),
+#ifdef TxJS_HOIST
+    txTop(NULL),
+    hasTxTop(false),
+#endif
     blockChain(prs->context),
     staticLevel(staticLevel),
     parenDepth(0),

