--- mozilla-esr17-orig/uriloader/base/nsURILoader.h	2013-02-15 14:59:57.000000000 -0600
+++ mozilla-esr17/uriloader/base/nsURILoader.h	2014-06-23 22:02:00.861039721 -0500
@@ -15,6 +15,9 @@
 #include "nsString.h"
 #include "nsIWeakReference.h"
 #include "mozilla/Attributes.h"
+// TxJS begin
+#include "jsapi.h"
+// TxJS end
 
 #ifdef MOZ_LOGGING
 // Uncomment the next line to force logging on in release builds
@@ -59,6 +62,10 @@
 #endif
   
   friend class nsDocumentOpenInfo;
+#ifdef TxJS
+public:
+  JS::Value mIntrospector;
+#endif
 };
 
 #endif /* nsURILoader_h__ */
