--- ff-orig/mozilla-23.0.1/content/base/src/nsFrameLoader.cpp	2013-08-14 08:49:18.000000000 -0500
+++ ff-dev/mozilla-23.0.1/content/base/src/nsFrameLoader.cpp	2015-01-21 08:54:51.462392616 -0600
@@ -289,6 +289,9 @@
   , mRemoteBrowser(nullptr)
   , mRenderMode(RENDER_MODE_DEFAULT)
   , mEventMode(EVENT_MODE_NORMAL_DISPATCH)
+#ifdef TxJS
+  , mIntrospector(JS::UndefinedValue())
+#endif
 {
   ResetPermissionManagerStatus();
 }
@@ -480,6 +483,9 @@
   // Kick off the load...
   bool tmpState = mNeedsAsyncDestroy;
   mNeedsAsyncDestroy = true;
+#ifdef TxJS
+  mDocShell->SetIntrospector(mIntrospector);
+#endif
   rv = mDocShell->LoadURI(mURIToLoad, loadInfo, flags, false);
   mNeedsAsyncDestroy = tmpState;
   mURIToLoad = nullptr;

