--- ff-orig/mozilla-23.0.1/js/src/frontend/TokenStream.cpp	2013-08-14 08:49:29.000000000 -0500
+++ ff-dev/mozilla-23.0.1/js/src/frontend/TokenStream.cpp	2015-01-21 08:54:52.062392599 -0600
@@ -1835,6 +1835,9 @@
       case TOK_MULASSIGN:       return "TOK_MULASSIGN";
       case TOK_DIVASSIGN:       return "TOK_DIVASSIGN";
       case TOK_MODASSIGN:       return "TOK_MODASSIGN";
+#ifdef TxJS
+      case TOK_TX:              return "TOK_TX";
+#endif
       case TOK_EXPORT:          return "TOK_EXPORT";
       case TOK_IMPORT:          return "TOK_IMPORT";
       case TOK_LIMIT:           break;

