--- ff-orig/mozilla-23.0.1/js/src/jsapi.h	2013-08-14 08:49:31.000000000 -0500
+++ ff-dev/mozilla-23.0.1/js/src/jsapi.h	2015-01-21 08:54:52.118392597 -0600
@@ -1403,7 +1403,14 @@
  */
 #define JSFUN_GENERIC_NATIVE   0x800
 
+#ifdef TxJS
+#define JSFUN_TX_NOSUSPEND     0x1000   /* indicates that a call to this
+                                           function should not suspend
+                                           an active transaction */
+#define JSFUN_FLAGS_MASK       0x1e00   /* | of all the JSFUN_* flags */
+#else
 #define JSFUN_FLAGS_MASK       0xe00    /* | of all the JSFUN_* flags */
+#endif
 
 /*
  * The first call to JS_CallOnce by any thread in a process will call 'func'.
@@ -1436,6 +1443,11 @@
 extern JS_PUBLIC_API(JSString *)
 JS_GetEmptyString(JSRuntime *rt);
 
+#ifdef TxJS
+extern JS_PUBLIC_API(jsval)
+JS_GetIntrospector(JSContext *cx);
+#endif
+
 /*
  * Format is a string of the following characters (spaces are insignificant),
  * specifying the tabulated type conversions:
@@ -3655,6 +3667,7 @@
  * Security protocol.
  */
 struct JSPrincipals {
+
     /* Don't call "destroy"; use reference counting macros below. */
     int refcount;
 
@@ -3880,6 +3893,9 @@
     bool forEval;
     bool noScriptRval;
     bool selfHostingMode;
+#ifdef TxJS
+    Value introspector;
+#endif
     bool userBit;
     enum SourcePolicy {
         NO_SOURCE,

