--- ff-orig/mozilla-23.0.1/js/src/jsversion.h	2013-08-14 08:49:31.000000000 -0500
+++ ff-dev/mozilla-23.0.1/js/src/jsversion.h	2015-01-21 08:54:52.394392589 -0600
@@ -39,6 +39,9 @@
 #define JS_HAS_GENERATOR_EXPRS  1       /* has (expr for (lhs in iterable)) */
 #define JS_HAS_EXPR_CLOSURES    1       /* has function (formals) listexpr */
 
+/* Enable transactions */
+#define TxJS 1
+
 /* Support for JS_NewGlobalObject. */
 #define JS_HAS_NEW_GLOBAL_OBJECT        1
 

