--- ff-orig/mozilla-23.0.1/js/src/vm/CommonPropertyNames.h	2013-08-14 08:49:33.000000000 -0500
+++ ff-dev/mozilla-23.0.1/js/src/vm/CommonPropertyNames.h	2015-01-21 08:54:52.462392587 -0600
@@ -157,6 +157,17 @@
     macro(string, string, "string") \
     macro(number, number, "number") \
     macro(boolean, boolean, "boolean") \
+    /* TxJS begin */ \
+    macro(subsumedBy, subsumedBy, "subsumedBy") \
+    macro(read, read, "read") \
+    macro(write, write, "write") \
+    macro(forin, forin, "forin") \
+    macro(type, type, "type") \
+    macro(obj, obj, "obj") \
+    macro(id, id, "id") \
+    macro(argc, argc, "argc") \
+    macro(args, args, "args") \
+    /* TxJS end */ \
     macro(null, null, "null")
 
 #endif /* CommonPropertyNames_h__ */

