--- ff-orig/mozilla-23.0.1/js/src/jsopcode.tbl	2013-08-14 08:49:31.000000000 -0500
+++ ff-dev/mozilla-23.0.1/js/src/jsopcode.tbl	2015-01-21 08:54:52.310392592 -0600
@@ -545,3 +545,10 @@
 
 /* Notes the point at which a value is pushed as an argument. */
 OPDEF(JSOP_NOTEARG,       228, "notearg",      NULL,  1,  0,  0,  0,  JOF_BYTE)
+
+// TxJS begin
+// %%% Guarding with a preprocessor directive causes
+// %%% inconsistencies.
+OPDEF(JSOP_BEGIN_TX,      229, "begin_tx",     NULL,  3,  1,  0,  0,  JOF_UINT16)
+OPDEF(JSOP_END_TX,        230, "end_tx",       NULL,  1,  0,  1,  0,  JOF_BYTE|JOF_INVOKE)
+// TxJS end

