--- ff-orig/mozilla-23.0.1/js/src/frontend/SharedContext.h	2013-08-14 08:49:29.000000000 -0500
+++ ff-dev/mozilla-23.0.1/js/src/frontend/SharedContext.h	2015-01-21 08:54:52.062392599 -0600
@@ -13,6 +13,9 @@
 #include "jsscript.h"
 #include "jsprvtd.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "builtin/Module.h"
 #include "frontend/ParseMaps.h"
@@ -257,6 +260,9 @@
     STMT_IF,                    /* if (then) statement */
     STMT_ELSE,                  /* else clause of if statement */
     STMT_SEQ,                   /* synthetic sequence of statements */
+#ifdef TxJS
+    STMT_TX,                    /* transaction block */
+#endif
     STMT_BLOCK,                 /* compound statement: { s1[;... sN] } */
     STMT_SWITCH,                /* switch statement */
     STMT_WITH,                  /* with statement */

