--- ff-orig/mozilla-29.0.1/docshell/base/nsDocShell.cpp	2014-05-06 17:55:21.000000000 -0500
+++ ff-dev/mozilla-29.0.1/docshell/base/nsDocShell.cpp	2015-01-25 16:09:23.338992436 -0600
@@ -846,6 +846,9 @@
     mDefaultLoadFlags(nsIRequest::LOAD_NORMAL),
     mFrameType(eFrameTypeRegular),
     mOwnOrContainingAppId(nsIScriptSecurityManager::UNKNOWN_APP_ID),
+#ifdef TxJS
+    mIntrospector(JS::UndefinedValue()),
+#endif
     mParentCharsetSource(0)
 {
     mHistoryID = ++gDocshellIDCounter;
@@ -1806,6 +1809,20 @@
         originIsFile && targetIsFile;
 }
 
+#ifdef TxJS
+NS_IMETHODIMP
+nsDocShell::GetIntrospector(JS::Value *aIntrospector) {
+    *aIntrospector = mIntrospector;
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsDocShell::SetIntrospector(const JS::Value & aIntrospector) {
+    mIntrospector = aIntrospector;
+    return NS_OK;
+}
+#endif
+
 NS_IMETHODIMP
 nsDocShell::GetEldestPresContext(nsPresContext** aPresContext)
 {
@@ -9623,6 +9640,10 @@
 
     uriLoader = do_GetService(NS_URI_LOADER_CONTRACTID, &rv);
     if (NS_FAILED(rv)) return rv;
+#ifdef TxJS
+    JS::Value ispect;
+    GetIntrospector(&ispect);
+#endif
 
     nsLoadFlags loadFlags = mDefaultLoadFlags;
     if (aFirstParty) {
@@ -9881,6 +9902,9 @@
         // Allow execution against our context if the principals match
         scriptChannel->
             SetExecutionPolicy(nsIScriptChannel::EXECUTE_NORMAL);
+#ifdef TxJS
+        scriptChannel->SetIntrospector(ispect);
+#endif
     }
 
     if (aIsNewWindowTarget) {
@@ -9896,7 +9920,7 @@
     if (timedChannel) {
         timedChannel->SetTimingEnabled(true);
     }
-
+    static_cast<nsURILoader*>(uriLoader.get())->mIntrospector = ispect;
     rv = DoChannelLoad(channel, uriLoader, aBypassClassifier);
 
     //

