--- ff-orig/mozilla-26.0/content/base/src/nsFrameLoader.cpp	2013-12-05 10:07:23.000000000 -0600
+++ ff-dev/mozilla-26.0/content/base/src/nsFrameLoader.cpp	2015-01-24 14:47:27.533662355 -0600
@@ -291,6 +291,9 @@
   , mChildID(0)
   , mRenderMode(RENDER_MODE_DEFAULT)
   , mEventMode(EVENT_MODE_NORMAL_DISPATCH)
+#ifdef TxJS
+  , mIntrospector(JS::UndefinedValue())
+#endif
 {
   ResetPermissionManagerStatus();
 }
@@ -505,6 +508,9 @@
   // Kick off the load...
   bool tmpState = mNeedsAsyncDestroy;
   mNeedsAsyncDestroy = true;
+#ifdef TxJS
+  mDocShell->SetIntrospector(mIntrospector);
+#endif
   nsCOMPtr<nsIURI> uriToLoad = mURIToLoad;
   rv = mDocShell->LoadURI(uriToLoad, loadInfo, flags, false);
   mNeedsAsyncDestroy = tmpState;

