--- ff-orig/mozilla-26.0/docshell/base/nsIDocShell.idl	2013-12-05 10:07:26.000000000 -0600
+++ ff-dev/mozilla-26.0/docshell/base/nsIDocShell.idl	2015-01-24 14:47:27.689662350 -0600
@@ -180,6 +180,10 @@
    */ 
   void setCurrentURI(in nsIURI aURI);
 
+  // TxJS begin
+  attribute jsval introspector;
+  // TxJS end
+
   /**
    * Notify the associated content viewer and all child docshells that they are
    * about to be hidden.  If |isUnload| is true, then the document is being

