--- ff-orig/mozilla-26.0/dom/base/nsIScriptContext.h	2013-12-05 10:07:27.000000000 -0600
+++ ff-dev/mozilla-26.0/dom/base/nsIScriptContext.h	2015-01-24 14:53:07.669652410 -0600
@@ -61,12 +61,22 @@
    * @param aOffThreadToken if specified, the result of compiling the script
    *                        on another thread.
    */
+#ifdef TxJS
   virtual nsresult EvaluateString(const nsAString& aScript,
                                   JS::Handle<JSObject*> aScopeObject,
                                   JS::CompileOptions& aOptions,
                                   bool aCoerceToString,
                                   JS::Value* aRetValue,
+                                  const JS::Value &ispect,
                                   void **aOffThreadToken = nullptr) = 0;
+#else
+  virtual nsresult EvaluateString(const nsAString& aScript,
+                                  JS::Handle<JSObject*> aScopeObject,
+                                  JS::CompileOptions& aOptions,
+                                  bool aCoerceToString,
+                                  JS::Value* aRetValue,
+                                  void **aOffThreadToken = nullptr) = 0;
+#endif
 
   /**
    * Bind an already-compiled event handler function to the given

