--- ff-orig/mozilla-26.0/js/src/vm/CommonPropertyNames.h	2013-12-05 10:07:40.000000000 -0600
+++ ff-dev/mozilla-26.0/js/src/vm/CommonPropertyNames.h	2015-01-24 14:47:28.249662334 -0600
@@ -165,6 +165,17 @@
     macro(string, string, "string") \
     macro(number, number, "number") \
     macro(boolean, boolean, "boolean") \
+    /* TxJS begin */ \
+    macro(subsumedBy, subsumedBy, "subsumedBy") \
+    macro(read, read, "read") \
+    macro(write, write, "write") \
+    macro(forin, forin, "forin") \
+    macro(type, type, "type") \
+    macro(obj, obj, "obj") \
+    macro(id, id, "id") \
+    macro(argc, argc, "argc") \
+    macro(args, args, "args") \
+    /* TxJS end */ \
     macro(null, null, "null")
 
 #endif /* vm_CommonPropertyNames_h */

