--- ff-orig/mozilla-26.0/js/src/jsversion.h	2013-12-05 10:07:37.000000000 -0600
+++ ff-dev/mozilla-26.0/js/src/jsversion.h	2015-01-24 14:47:28.193662335 -0600
@@ -26,6 +26,9 @@
 #define JS_HAS_GENERATOR_EXPRS  1       /* has (expr for (lhs in iterable)) */
 #define JS_HAS_EXPR_CLOSURES    1       /* has function (formals) listexpr */
 
+/* Enable transactions */
+#define TxJS 1
+
 /* Support for JS_NewGlobalObject. */
 #define JS_HAS_NEW_GLOBAL_OBJECT        1
 

