--- ff-orig/mozilla-26.0/js/src/jsobj.cpp	2013-12-05 10:07:37.000000000 -0600
+++ ff-dev/mozilla-26.0/js/src/jsobj.cpp	2015-01-24 15:48:07.205555937 -0600
@@ -24,6 +24,10 @@
 #include "jsfun.h"
 #include "jsgc.h"
 #include "jsiter.h"
+#ifdef TxJS
+#include "jstransaction.h"
+#include "jsjam.h"
+#endif
 #include "jsnum.h"
 #include "jsopcode.h"
 #include "jsprf.h"
@@ -1596,6 +1600,17 @@
 js::CreateThisForFunction(JSContext *cx, HandleObject callee, bool newType)
 {
     RootedValue protov(cx);
+#ifdef TxJS
+    TxContext *tx = cx->getRunningTx();
+    if (tx) {
+        jsid protoId = AtomToId(cx->names().classPrototype);
+        if (tx->getProperty(cx, callee, protoId, &protov)) {
+        } else {
+            if (!JSObject::getProperty(cx, callee, callee, cx->names().classPrototype, &protov))
+                return NULL;
+        }
+    } else
+#endif
     if (!JSObject::getProperty(cx, callee, callee, cx->names().classPrototype, &protov))
         return NULL;
     JSObject *proto;

