--- ff-orig/mozilla-26.0/js/src/jit/Ion.h	2013-12-05 10:07:37.000000000 -0600
+++ ff-dev/mozilla-26.0/js/src/jit/Ion.h	2015-01-24 14:47:27.969662342 -0600
@@ -364,6 +364,10 @@
 static inline bool
 IsIonEnabled(JSContext *cx)
 {
+#ifdef TxJS
+    if (cx->getRunningTx())
+        return false;
+#endif
     return cx->hasOption(JSOPTION_ION) &&
         cx->hasOption(JSOPTION_BASELINE) &&
         cx->typeInferenceEnabled();

