--- ff-orig/mozilla-24.8.1esr/js/src/frontend/TokenStream.cpp	2014-09-23 21:53:35.000000000 -0500
+++ ff-dev/mozilla-24.8.1esr/js/src/frontend/TokenStream.cpp	2015-01-23 17:27:18.430951617 -0600
@@ -1896,6 +1896,9 @@
       case TOK_MULASSIGN:       return "TOK_MULASSIGN";
       case TOK_DIVASSIGN:       return "TOK_DIVASSIGN";
       case TOK_MODASSIGN:       return "TOK_MODASSIGN";
+#ifdef TxJS
+      case TOK_TX:              return "TOK_TX";
+#endif
       case TOK_EXPORT:          return "TOK_EXPORT";
       case TOK_IMPORT:          return "TOK_IMPORT";
       case TOK_LIMIT:           break;

