--- ff-orig/mozilla-24.8.1esr/js/src/jsstr.cpp	2014-09-23 21:53:36.000000000 -0500
+++ ff-dev/mozilla-24.8.1esr/js/src/jsstr.cpp	2015-01-23 17:27:18.722951609 -0600
@@ -3569,7 +3569,11 @@
 }
 
 static const JSFunctionSpec string_static_methods[] = {
+#ifdef TxJS
+    JS_FN("fromCharCode", js::str_fromCharCode, 1, JSFUN_TX_NOSUSPEND),
+#else
     JS_FN("fromCharCode", js::str_fromCharCode, 1, 0),
+#endif
 
     // This must be at the end because of bug 853075: functions listed after
     // self-hosted methods aren't available in self-hosted code.

