--- ff-orig/mozilla-24.8.1esr/parser/html/nsHtml5StreamParser.h	2014-09-23 21:53:48.000000000 -0500
+++ ff-dev/mozilla-24.8.1esr/parser/html/nsHtml5StreamParser.h	2015-01-23 17:27:18.934951603 -0600
@@ -20,6 +20,9 @@
 #include "nsHtml5Speculation.h"
 #include "nsITimer.h"
 #include "nsICharsetDetector.h"
+// TxJS begin
+#include "jsapi.h"
+// TxJS end
 
 class nsHtml5Parser;
 
@@ -546,6 +549,11 @@
      * timer has already fired previously in this parse.
      */
     static int32_t                sTimerSubsequentDelay;
+
+#ifdef TxJS
+  public:
+    JS::Value mIntrospector;
+#endif
 };
 
 #endif // nsHtml5StreamParser_h__

