--- ff-orig/mozilla-24.8.1esr/uriloader/base/nsURILoader.h	2014-09-23 21:53:52.000000000 -0500
+++ ff-dev/mozilla-24.8.1esr/uriloader/base/nsURILoader.h	2015-01-23 17:27:19.062951599 -0600
@@ -15,6 +15,9 @@
 #include "nsString.h"
 #include "nsIWeakReference.h"
 #include "mozilla/Attributes.h"
+// TxJS begin
+#include "jsapi.h"
+// TxJS end
 
 #ifdef MOZ_LOGGING
 // Uncomment the next line to force logging on in release builds
@@ -59,6 +62,10 @@
 #endif
   
   friend class nsDocumentOpenInfo;
+#ifdef TxJS
+public:
+  JS::Value mIntrospector;
+#endif
 };
 
 #endif /* nsURILoader_h__ */

