--- ff-orig/mozilla-22.0/content/base/src/nsObjectLoadingContent.h	2013-06-18 06:01:13.000000000 -0500
+++ ff-dev/mozilla-22.0/content/base/src/nsObjectLoadingContent.h	2015-01-21 09:13:56.746359130 -0600
@@ -488,7 +488,9 @@
     // as a root for other resource requests.
     nsCOMPtr<nsIURI>            mBaseURI;
 
-
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
 
     // Type of the currently-loaded content.
     ObjectType                  mType           : 8;

