--- ff-orig/mozilla-22.0/content/base/src/nsScriptLoader.cpp	2013-06-18 06:01:13.000000000 -0500
+++ ff-dev/mozilla-22.0/content/base/src/nsScriptLoader.cpp	2015-01-20 21:57:06.483546505 -0600
@@ -48,6 +48,9 @@
 
 #include "mozilla/CORSMode.h"
 #include "mozilla/Attributes.h"
+#ifdef TxJS
+#include "nsDocLoader.h"
+#endif
 
 #ifdef PR_LOGGING
 static PRLogModuleInfo* gCspPRLog;
@@ -70,6 +73,9 @@
       mIsInline(true),
       mJSVersion(aVersion),
       mLineNo(1),
+#ifdef TxJS
+      mIntrospector(JS::UndefinedValue()),
+#endif
       mCORSMode(aCORSMode)
   {
   }
@@ -99,6 +105,9 @@
   nsCOMPtr<nsIPrincipal> mOriginPrincipal;
   int32_t mLineNo;
   const CORSMode mCORSMode;
+#ifdef TxJS
+  JS::Value mIntrospector;
+#endif
 };
 
 // The nsScriptLoadRequest is passed as the context to necko, and thus
@@ -115,6 +124,9 @@
     mBlockerCount(0),
     mEnabled(true),
     mDeferEnabled(false),
+#ifdef TxJS
+    mIntrospector(JS::UndefinedValue()),
+#endif
     mDocumentParsingDone(false)
 {
   // enable logging for CSP
@@ -525,6 +537,14 @@
     request->mJSVersion = version;
 
     if (aElement->GetScriptAsync()) {
+#ifdef TxJS
+      JSContext *cx = nullptr;
+      cx = globalObject->GetScriptContext()->GetNativeContext();
+      JS::Value ispect = JS_GetIntrospector(cx);
+      if (!ispect.isUndefined()) {
+        request->mIntrospector = ispect;
+      }
+#endif
       mAsyncRequests.AppendElement(request);
       if (!request->mLoading) {
         // The script is available already. Run it ASAP when the event
@@ -652,6 +672,24 @@
   request->mURI = mDocument->GetDocumentURI();
   request->mLineNo = aElement->GetScriptLineNumber();
 
+#ifdef TxJS
+  /*
+  JSContext *cx = nullptr;
+  cx = globalObject->GetScriptContext()->GetNativeContext();
+  JS::Value ispect = JS_GetIntrospector(cx);
+  if (ispect.isUndefined()) {
+    nsIDocShell *ds = ((nsPIDOMWindow*)globalObject)->GetDocShell();
+    nsDocLoader *dl = (nsDocLoader*)ds;
+    if (false && !dl->introspector.isUndefined()) {
+      ispect = dl->introspector;
+    }
+  }
+  */
+  if (!mIntrospector.isUndefined()) {
+    request->mIntrospector = mIntrospector;
+  }
+#endif
+
   if (aElement->GetParserCreated() == FROM_PARSER_XSLT &&
       (!ReadyToExecuteScripts() || !mXSLTRequests.IsEmpty())) {
     // Need to maintain order for XSLT-inserted scripts
@@ -848,8 +886,14 @@
     if (aRequest->mOriginPrincipal) {
       options.setOriginPrincipals(nsJSPrincipals::get(aRequest->mOriginPrincipal));
     }
+#ifdef TxJS
+    rv = context->EvaluateString(aScript, *globalObject->GetGlobalJSObject(),
+                                 options, /* aCoerceToString = */ false,
+                                 nullptr, aRequest->mIntrospector);
+#else
     rv = context->EvaluateString(aScript, *globalObject->GetGlobalJSObject(),
                                  options, /* aCoerceToString = */ false, nullptr);
+#endif
   }
 
   // Put the old script back in case it wants to do anything else.

