--- ff-orig/mozilla-22.0/dom/base/nsJSEnvironment.h	2013-06-18 06:01:16.000000000 -0500
+++ ff-dev/mozilla-22.0/dom/base/nsJSEnvironment.h	2015-01-21 09:13:56.938359125 -0600
@@ -44,6 +44,18 @@
 
   virtual nsIScriptObjectPrincipal* GetObjectPrincipal();
 
+#ifdef TxJS
+  virtual nsresult EvaluateString(const nsAString& aScript,
+                                  JSObject* aScopeObject,
+                                  nsIPrincipal *principal,
+                                  nsIPrincipal *originPrincipal,
+                                  const char *aURL,
+                                  uint32_t aLineNo,
+                                  JSVersion aVersion,
+                                  nsAString *aRetValue,
+                                  bool* aIsUndefined,
+                                  const JS::Value &ispect);
+#endif
   virtual nsresult EvaluateString(const nsAString& aScript,
                                   JSObject& aScopeObject,
                                   JS::CompileOptions &aOptions,

