--- ff-orig/mozilla-22.0/js/src/methodjit/Compiler.cpp	2013-06-18 06:01:26.000000000 -0500
+++ ff-dev/mozilla-22.0/js/src/methodjit/Compiler.cpp	2015-01-20 21:57:07.171546485 -0600
@@ -992,6 +992,10 @@
   checkOutput:
     if (!cx->methodJitEnabled)
         return Compile_Abort;
+#ifndef TxJS_LOCAL
+    if (cx->getRunningTx())
+        return Compile_Abort;
+#endif
 
     /*
      * If SPS (profiling) is enabled, then the emitted instrumentation has to be
@@ -3294,6 +3298,20 @@
           }
           END_CASE(JSOP_DEBUGGER)
 
+#ifdef TxJS
+          BEGIN_CASE(JSOP_BEGIN_TX)
+          {
+            fprintf(stderr, "JIT entering JSOP_BEGIN_TX\n");
+          }
+          END_CASE(JSOP_BEGIN_TX)
+
+          BEGIN_CASE(JSOP_END_TX)
+          {
+            fprintf(stderr, "JIT entering JSOP_END_TX\n");
+          }
+          END_CASE(JSOP_END_TX)
+#endif
+
           default:
             JS_NOT_REACHED("Opcode not implemented");
         }

