--- ff-orig/mozilla-22.0/js/src/jsinterp.h	2013-06-18 06:01:26.000000000 -0500
+++ ff-dev/mozilla-22.0/js/src/jsinterp.h	2015-01-21 09:13:57.198359117 -0600
@@ -232,6 +232,20 @@
 extern bool
 HasInstance(JSContext *cx, HandleObject obj, HandleValue v, JSBool *bp);
 
+#ifdef TxJS
+extern bool 
+BeginTx(JSContext *cx, const js::Value &tx_handler);
+
+extern bool 
+IntrospectTx(JSContext *cx, TxContext *tx, bool isEnd, HandleScript script, int *path, bool *newType);
+
+extern bool
+TxSubsumed(JSContext *cx, const js::Value &ispect);
+
+extern bool
+SuspendCall(JSContext *cx, TxContext *tx, CallArgs &args, bool *construct);
+#endif
+
 /*
  * A linked list of the |FrameRegs regs;| variables belonging to all
  * js::Interpret C++ frames on this thread's stack.

