--- ff-orig/mozilla-22.0/js/src/frontend/SharedContext.h	2013-06-18 06:01:24.000000000 -0500
+++ ff-dev/mozilla-22.0/js/src/frontend/SharedContext.h	2015-01-21 09:13:57.070359121 -0600
@@ -14,6 +14,9 @@
 #include "jsscript.h"
 #include "jsprvtd.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "builtin/Module.h"
 #include "frontend/ParseMaps.h"
@@ -258,6 +261,9 @@
     STMT_IF,                    /* if (then) statement */
     STMT_ELSE,                  /* else clause of if statement */
     STMT_SEQ,                   /* synthetic sequence of statements */
+#ifdef TxJS
+    STMT_TX,                    /* transaction block */
+#endif
     STMT_BLOCK,                 /* compound statement: { s1[;... sN] } */
     STMT_SWITCH,                /* switch statement */
     STMT_WITH,                  /* with statement */

