--- ff-orig/mozilla-22.0/js/src/jsprototypes.h	2013-06-18 06:01:26.000000000 -0500
+++ ff-dev/mozilla-22.0/js/src/jsprototypes.h	2015-01-21 09:13:57.266359115 -0600
@@ -57,5 +57,7 @@
     macro(DataView,              35,     js_InitTypedArrayClasses) \
     macro(ParallelArray,         36,     js_InitParallelArrayClass) \
     macro(Intl,                  37,     js_InitIntlClass) \
+    macro(Tx,                    38,     js_InitTxClass) \
+    macro(JAM,                   39,     js_InitJAMClass) \
 
 #endif /* jsprototypes_h___ */

