--- ff-orig/mozilla-22.0/js/src/frontend/Parser.h	2013-06-18 06:01:24.000000000 -0500
+++ ff-dev/mozilla-22.0/js/src/frontend/Parser.h	2015-01-20 22:23:33.539500102 -0600
@@ -17,6 +17,9 @@
 #include "jsatom.h"
 #include "jsscript.h"
 #include "jswin.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "frontend/FoldConstants.h"
 #include "frontend/FullParseHandler.h"
@@ -63,6 +66,10 @@
 
     StmtInfoPC      *topStmt;       /* top of statement info stack */
     StmtInfoPC      *topScopeStmt;  /* top lexical scope statement */
+#ifdef TxJS_HOIST
+    StmtInfoPC      *txTop;         /* top statement to use for tx */
+    bool             hasTxTop;
+#endif
     Rooted<StaticBlockObject *> blockChain;
                                     /* compile time block scope chain */
 
@@ -415,6 +422,10 @@
     Node expr();
     Node assignExpr();
     Node assignExprWithoutYield(unsigned err);
+#ifdef TxJS
+    Node txStmt();
+    Node txExpr();
+#endif
     Node condExpr1();
     Node orExpr1();
     Node unaryExpr();
@@ -430,6 +441,9 @@
 
     Node functionDef(HandlePropertyName name, const TokenStream::Position &start,
                      size_t startOffset, FunctionType type, FunctionSyntaxKind kind);
+#ifdef TxJS
+    bool txArguments(Node txpn);
+#endif
     bool functionArgsAndBody(Node pn, HandleFunction fun, HandlePropertyName funName,
                              size_t startOffset, FunctionType type, FunctionSyntaxKind kind,
                              bool strict, bool *becameStrict = NULL);

