--- ff-orig/mozilla-22.0/js/src/frontend/TokenStream.cpp	2013-06-18 06:01:24.000000000 -0500
+++ ff-dev/mozilla-22.0/js/src/frontend/TokenStream.cpp	2015-01-20 21:57:06.831546495 -0600
@@ -1827,6 +1827,9 @@
       case TOK_MULASSIGN:       return "TOK_MULASSIGN";
       case TOK_DIVASSIGN:       return "TOK_DIVASSIGN";
       case TOK_MODASSIGN:       return "TOK_MODASSIGN";
+#ifdef TxJS
+      case TOK_TX:              return "TOK_TX";
+#endif
       case TOK_EXPORT:          return "TOK_EXPORT";
       case TOK_IMPORT:          return "TOK_IMPORT";
       case TOK_LIMIT:           break;

