--- ff-orig/mozilla-22.0/js/src/Makefile.in	2013-06-18 06:01:23.000000000 -0500
+++ ff-dev/mozilla-22.0/js/src/Makefile.in	2015-01-21 09:13:57.310359114 -0600
@@ -69,6 +69,7 @@
 		jsinfer.cpp \
 		jsinterp.cpp \
 		jsiter.cpp \
+    jsjam.cpp \
 		jslog2.cpp \
 		jsmath.cpp \
 		jsmemorymetrics.cpp \
@@ -86,6 +87,9 @@
 		jsreflect.cpp \
 		jsscript.cpp \
 		jsstr.cpp \
+    jstransaction.cpp \
+    jstx.cpp \
+    jstxaction.cpp \
 		jstypedarray.cpp \
 		jsutil.cpp \
 		jswatchpoint.cpp \

