--- ff-orig/mozilla-22.0/js/src/jsscript.cpp	2013-06-18 06:01:26.000000000 -0500
+++ ff-dev/mozilla-22.0/js/src/jsscript.cpp	2015-01-21 09:13:57.274359115 -0600
@@ -1682,6 +1682,10 @@
     script->sourceEnd = bufEnd;
     script->userBit = options.userBit;
 
+#ifdef TxJS
+    script->setIntrospector(UndefinedValue());
+#endif
+
     return script;
 }
 
@@ -2362,6 +2366,9 @@
     dst->hasSingletons = src->hasSingletons;
     dst->isGenerator = src->isGenerator;
     dst->isGeneratorExp = src->isGeneratorExp;
+#ifdef TxJS
+    dst->setIntrospector(src->getIntrospector());
+#endif
 
     /* Copy over hints. */
     dst->shouldCloneAtCallsite = src->shouldCloneAtCallsite;

