--- ff-orig/mozilla-22.0/js/src/frontend/TokenStream.h	2013-06-18 06:01:24.000000000 -0500
+++ ff-dev/mozilla-22.0/js/src/frontend/TokenStream.h	2015-01-21 09:13:57.070359121 -0600
@@ -21,6 +21,9 @@
 #include "jsopcode.h"
 #include "jsprvtd.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "js/Vector.h"
 
@@ -150,7 +153,9 @@
     TOK_DIVASSIGN,
     TOK_MODASSIGN,
     TOK_ASSIGNMENT_LAST = TOK_MODASSIGN,
-
+#ifdef TxJS
+    TOK_TX,
+#endif
     TOK_LIMIT                      /* domain size */
 };
 

