--- ff-orig/mozilla-22.0/parser/html/nsHtml5TreeOpExecutor.cpp	2013-06-18 06:01:38.000000000 -0500
+++ ff-dev/mozilla-22.0/parser/html/nsHtml5TreeOpExecutor.cpp	2015-01-21 09:13:57.522359108 -0600
@@ -68,6 +68,9 @@
 
 nsHtml5TreeOpExecutor::nsHtml5TreeOpExecutor(bool aRunsToCompletion)
 {
+#ifdef TxJS
+  mIntrospector = JS::UndefinedValue();
+#endif
   mRunsToCompletion = aRunsToCompletion;
   mPreloadedURLs.Init(23); // Mean # of preloadable resources per page on dmoz
   // zeroing operator new for everything else
@@ -762,6 +765,15 @@
 
   NS_ASSERTION(aScriptElement, "No script to run");
   nsCOMPtr<nsIScriptElement> sele = do_QueryInterface(aScriptElement);
+#ifdef TxJS
+  if (!mIntrospector.isUndefined()) {
+    nsCOMPtr<nsIContent> cont =
+      do_QueryInterface((nsIScriptElement*) sele);
+    nsIDocument* ownerDoc = cont->OwnerDoc();
+    nsRefPtr<nsScriptLoader> loader = ownerDoc->ScriptLoader();
+    loader->mIntrospector = mIntrospector;
+  }
+#endif
   
   if (!mParser) {
     NS_ASSERTION(sele->IsMalformed(), "Script wasn't marked as malformed.");

