--- ff-orig/mozilla-22.0/parser/html/nsHtml5TreeOpExecutor.h	2013-06-18 06:01:38.000000000 -0500
+++ ff-dev/mozilla-22.0/parser/html/nsHtml5TreeOpExecutor.h	2015-01-20 21:57:07.591546473 -0600
@@ -25,6 +25,10 @@
 #include "nsTHashtable.h"
 #include "nsHashKeys.h"
 #include "mozilla/LinkedList.h"
+// TxJS begin
+// %%% Needed?
+#include "jsapi.h"
+// TxJS end
 
 class nsHtml5Parser;
 class nsHtml5TreeBuilder;
@@ -52,6 +56,10 @@
     NS_DECL_ISUPPORTS_INHERITED
     NS_DECL_CYCLE_COLLECTION_CLASS_INHERITED(nsHtml5TreeOpExecutor, nsContentSink)
 
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
+
   private:
     static bool        sExternalViewSource;
 #ifdef DEBUG_NS_HTML5_TREE_OP_EXECUTOR_FLUSH

