--- ff-orig/mozilla-30.0/content/base/src/nsObjectLoadingContent.h	2014-06-05 20:15:19.000000000 -0500
+++ ff-dev/mozilla-30.0/content/base/src/nsObjectLoadingContent.h	2015-01-25 17:34:26.134843239 -0600
@@ -537,7 +537,9 @@
     // as a root for other resource requests.
     nsCOMPtr<nsIURI>            mBaseURI;
 
-
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
 
     // Type of the currently-loaded content.
     ObjectType                  mType           : 8;

