--- ff-orig/mozilla-30.0/content/base/src/nsScriptLoader.h	2014-06-05 20:15:19.000000000 -0500
+++ ff-dev/mozilla-30.0/content/base/src/nsScriptLoader.h	2015-01-25 17:34:26.258843235 -0600
@@ -17,6 +17,9 @@
 #include "nsAutoPtr.h"
 #include "nsIDocument.h"
 #include "nsIStreamLoader.h"
+// TxJS begin
+#include "jsapi.h"
+// TxJS end
 
 class nsScriptLoadRequest;
 class nsIURI;
@@ -214,6 +217,9 @@
   nsresult ProcessOffThreadRequest(nsScriptLoadRequest *aRequest,
                                    void **aOffThreadToken);
 
+#ifdef TxJS
+  JS::Value mIntrospector;
+#endif
 private:
   /**
    * Unblocks the creator parser of the parser-blocking scripts.

