--- ff-orig/mozilla-30.0/docshell/base/nsDocShell.h	2014-06-05 20:15:20.000000000 -0500
+++ ff-dev/mozilla-30.0/docshell/base/nsDocShell.h	2015-01-25 17:34:26.710843222 -0600
@@ -767,6 +767,10 @@
     nsIDocShellTreeOwner *     mTreeOwner; // Weak Reference
     mozilla::dom::EventTarget* mChromeEventHandler; //Weak Reference
 
+#ifdef TxJS
+    JS::Value mIntrospector;
+#endif
+
     eCharsetReloadState        mCharsetReloadState;
 
     // Offset in the parent's child list.

