--- ff-orig/mozilla-30.0/js/src/jsapi.h	2014-06-05 20:15:28.000000000 -0500
+++ ff-dev/mozilla-30.0/js/src/jsapi.h	2015-01-25 17:34:28.294843176 -0600
@@ -937,7 +937,14 @@
  */
 #define JSFUN_GENERIC_NATIVE   0x800
 
+#ifdef TxJS
+#define JSFUN_TX_NOSUSPEND     0x1000   /* indicates that a call to this
+                                           function should not suspend
+                                           an active transaction */
+#define JSFUN_FLAGS_MASK       0x1e00   /* | of all the JSFUN_* flags */
+#else
 #define JSFUN_FLAGS_MASK       0xe00    /* | of all the JSFUN_* flags */
+#endif
 
 /*
  * The first call to JS_CallOnce by any thread in a process will call 'func'.
@@ -970,6 +977,11 @@
 extern JS_PUBLIC_API(JSString *)
 JS_GetEmptyString(JSRuntime *rt);
 
+#ifdef TxJS
+extern JS_PUBLIC_API(jsval)
+JS_GetIntrospector(JSContext *cx);
+#endif
+
 /*
  * Format is a string of the following characters (spaces are insignificant),
  * specifying the tabulated type conversions:
@@ -3438,6 +3450,9 @@
     bool forEval;
     bool noScriptRval;
     bool selfHostingMode;
+#ifdef TxJS
+    Value introspector;
+#endif
     bool canLazilyParse;
     bool strictOption;
     bool extraWarningsOption;

