--- ff-orig/mozilla-30.0/js/src/jsatom.h	2014-06-05 20:15:28.000000000 -0500
+++ ff-dev/mozilla-30.0/js/src/jsatom.h	2015-01-25 17:40:04.582833343 -0600
@@ -146,6 +146,9 @@
 extern const char js_in_str[];
 extern const char js_instanceof_str[];
 extern const char js_interface_str[];
+#ifdef TxJS
+extern const char js_introspect_str[];
+#endif
 extern const char js_new_str[];
 extern const char js_package_str[];
 extern const char js_private_str[];

