--- ff-orig/mozilla-30.0/js/src/jsscript.h	2014-06-05 20:15:28.000000000 -0500
+++ ff-dev/mozilla-30.0/js/src/jsscript.h	2015-01-25 17:34:30.426843113 -0600
@@ -70,6 +70,9 @@
     JSTRY_CATCH,
     JSTRY_FINALLY,
     JSTRY_ITER,
+#ifdef TxJS
+    JSTRY_TX,
+#endif
     JSTRY_LOOP
 } JSTryNoteKind;
 
@@ -703,6 +706,16 @@
     /* Persistent type information retained across GCs. */
     js::types::TypeScript *types;
 
+#ifdef TxJS
+  private:
+    js::Value introspector_;
+  public:
+    inline const js::Value getIntrospector() { return introspector_; }
+    inline void setIntrospector(const js::Value &ispect) {
+        introspector_ = ispect;
+    }
+#endif
+
   private:
     // This script's ScriptSourceObject, or a CCW thereof.
     //

