--- ff-orig/mozilla-30.0/js/src/jsprototypes.h	2014-06-05 20:15:28.000000000 -0500
+++ ff-dev/mozilla-30.0/js/src/jsprototypes.h	2015-01-25 17:42:53.954828391 -0600
@@ -102,6 +102,8 @@
 IF_BDATA(real,imaginary)(TypedObject,           38,     js_InitTypedObjectModuleObject,   OCLASP(TypedObjectModule)) \
     imaginary(GeneratorFunction,     39,     js_InitIteratorClasses, dummy) \
 IF_BDATA(real,imaginary)(SIMD,                  40,     js_InitSIMDClass, OCLASP(SIMD)) \
+    real(Tx,                    41,     js_InitTxClass,            CLASP(Tx)) \
+    real(JAM,                   42,     js_InitJAMClass,           CLASP(JAM)) \
 
 #define JS_FOR_EACH_PROTOTYPE(macro) JS_FOR_PROTOTYPES(macro,macro)
 

