--- ff-orig/mozilla-30.0/js/src/frontend/TokenStream.h	2014-06-05 20:15:27.000000000 -0500
+++ ff-dev/mozilla-30.0/js/src/frontend/TokenStream.h	2015-01-25 17:34:28.018843184 -0600
@@ -18,6 +18,9 @@
 
 #include "jscntxt.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "js/Vector.h"
 #include "vm/RegExpObject.h"
@@ -146,6 +149,9 @@
     TOK_MODASSIGN,
     TOK_ASSIGNMENT_LAST = TOK_MODASSIGN,
 
+#ifdef TxJS
+    TOK_TX,
+#endif
     TOK_LIMIT                      // domain size
 };
 

