--- ff-orig/mozilla-30.0/js/src/jsatom.cpp	2014-06-05 20:15:28.000000000 -0500
+++ ff-dev/mozilla-30.0/js/src/jsatom.cpp	2015-01-25 17:41:49.898830264 -0600
@@ -86,6 +86,9 @@
 const char js_in_str[]              = "in";
 const char js_instanceof_str[]      = "instanceof";
 const char js_interface_str[]       = "interface";
+#ifdef TxJS
+const char js_introspect_str[]      = "introspect";
+#endif
 const char js_new_str[]             = "new";
 const char js_package_str[]         = "package";
 const char js_private_str[]         = "private";

