--- ff-orig/mozilla-21.0/dom/base/nsJSEnvironment.cpp	2013-05-11 14:19:26.000000000 -0500
+++ ff-dev/mozilla-21.0/dom/base/nsJSEnvironment.cpp	2015-01-19 22:30:45.066630154 -0600
@@ -1235,7 +1235,7 @@
 
   return refcnt;
 }
-
+#ifdef TxJS
 nsresult
 nsJSContext::EvaluateString(const nsAString& aScript,
                             JSObject& aScopeObject,
@@ -1243,6 +1243,19 @@
                             bool aCoerceToString,
                             JS::Value* aRetValue)
 {
+  return EvaluateString(aScript, aScopeObject, aOptions,
+      aCoerceToString, aRetValue, JS::UndefinedValue());
+}
+
+nsresult
+nsJSContext::EvaluateString(const nsAString& aScript,
+                            JSObject& aScopeObject,
+                            JS::CompileOptions& aOptions,
+                            bool aCoerceToString,
+                            JS::Value* aRetValue,
+                            const JS::Value &ispect)
+{
+#endif
   SAMPLE_LABEL("JS", "EvaluateString");
   MOZ_ASSERT_IF(aOptions.versionSet, aOptions.version != JSVERSION_UNKNOWN);
   MOZ_ASSERT_IF(aCoerceToString, aRetValue);
@@ -1285,6 +1298,9 @@
     ++mExecuteDepth;
 
     js::RootedObject rootedScope(mContext, &aScopeObject);
+#ifdef TxJS
+    aOptions.introspector = ispect;
+#endif
     ok = JS::Evaluate(mContext, rootedScope, aOptions,
                       PromiseFlatString(aScript).get(),
                       aScript.Length(), aRetValue);

