--- ff-orig/mozilla-21.0/js/src/frontend/TokenStream.cpp	2013-05-11 14:19:33.000000000 -0500
+++ ff-dev/mozilla-21.0/js/src/frontend/TokenStream.cpp	2015-01-19 22:17:56.626652622 -0600
@@ -1792,6 +1792,9 @@
       case TOK_MULASSIGN:       return "TOK_MULASSIGN";
       case TOK_DIVASSIGN:       return "TOK_DIVASSIGN";
       case TOK_MODASSIGN:       return "TOK_MODASSIGN";
+#ifdef TxJS
+      case TOK_TX:              return "TOK_TX";
+#endif
       case TOK_EXPORT:          return "TOK_EXPORT";
       case TOK_IMPORT:          return "TOK_IMPORT";
       case TOK_LIMIT:           break;

