--- ff-orig/mozilla-21.0/js/src/jsarray.h	2013-05-11 14:19:34.000000000 -0500
+++ ff-dev/mozilla-21.0/js/src/jsarray.h	2015-01-19 22:17:56.762652618 -0600
@@ -14,6 +14,9 @@
 #include "jspubtd.h"
 #include "jsatom.h"
 #include "jsobj.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 namespace js {
 /* 2^32-2, inclusive */
@@ -137,4 +140,11 @@
 JSBool
 js_Array(JSContext *cx, unsigned argc, js::Value *vp);
 
+#ifdef TxJS
+extern JSBool
+js_SetArrayElement(JSContext *cx, js::HandleObject obj, double index, js::HandleValue vp);
+
+extern JSBool
+js_GetArrayElement(JSContext *cx, js::HandleObject obj, double index, JSBool *hole, js::MutableHandleValue vp);
+#endif
 #endif /* jsarray_h___ */

