--- ff-orig/mozilla-21.0/js/src/vm/Keywords.h	2013-05-11 14:19:36.000000000 -0500
+++ ff-dev/mozilla-21.0/js/src/vm/Keywords.h	2015-01-19 22:17:57.222652604 -0600
@@ -72,6 +72,9 @@
     macro(extends, extends, TOK_RESERVED, JSOP_NOP, JSVERSION_DEFAULT) \
     macro(import, import, TOK_RESERVED, JSOP_NOP, JSVERSION_DEFAULT) \
     macro(super, super, TOK_RESERVED, JSOP_NOP, JSVERSION_DEFAULT) \
+    /* TxJS begin */ \
+    macro(introspect, introspect_, TOK_TX, JSOP_BEGIN_TX, JSVERSION_DEFAULT) \
+    /* TxJS end */ \
     /* ES5 future reserved keywords in strict mode. */ \
     macro(implements, implements, TOK_STRICT_RESERVED, JSOP_NOP, JSVERSION_DEFAULT) \
     macro(interface, interface, TOK_STRICT_RESERVED, JSOP_NOP, JSVERSION_DEFAULT) \

