--- ff-orig/mozilla-21.0/js/src/frontend/ParseNode.h	2013-05-11 14:19:33.000000000 -0500
+++ ff-dev/mozilla-21.0/js/src/frontend/ParseNode.h	2015-01-19 22:17:56.582652623 -0600
@@ -184,6 +184,9 @@
 #define EMIT_ENUM(name) PNK_##name,
     FOR_EACH_PARSE_NODE_KIND(EMIT_ENUM)
 #undef EMIT_ENUM
+#ifdef TxJS
+    PNK_TX,
+#endif
     PNK_LIMIT, /* domain size */
     PNK_ASSIGNMENT_START = PNK_ASSIGN,
     PNK_ASSIGNMENT_LAST = PNK_MODASSIGN

