--- ff-orig/mozilla-31.0/content/base/src/nsFrameLoader.cpp	2014-07-16 20:45:08.000000000 -0500
+++ ff-dev/mozilla-31.0/content/base/src/nsFrameLoader.cpp	2015-01-25 19:38:31.970625535 -0600
@@ -285,6 +285,9 @@
   , mRenderMode(RENDER_MODE_DEFAULT)
   , mEventMode(EVENT_MODE_NORMAL_DISPATCH)
   , mPendingFrameSent(false)
+#ifdef TxJS
+  , mIntrospector(JS::UndefinedValue())
+#endif
 {
   ResetPermissionManagerStatus();
 }
@@ -560,6 +563,9 @@
   // Kick off the load...
   bool tmpState = mNeedsAsyncDestroy;
   mNeedsAsyncDestroy = true;
+#ifdef TxJS
+  mDocShell->SetIntrospector(mIntrospector);
+#endif
   nsCOMPtr<nsIURI> uriToLoad = mURIToLoad;
   rv = mDocShell->LoadURI(uriToLoad, loadInfo, flags, false);
   mNeedsAsyncDestroy = tmpState;

