--- ff-orig/mozilla-31.0/docshell/base/nsDocShell.cpp	2014-07-16 20:45:10.000000000 -0500
+++ ff-dev/mozilla-31.0/docshell/base/nsDocShell.cpp	2015-01-25 19:38:32.210625528 -0600
@@ -847,6 +847,9 @@
     mDefaultLoadFlags(nsIRequest::LOAD_NORMAL),
     mFrameType(eFrameTypeRegular),
     mOwnOrContainingAppId(nsIScriptSecurityManager::UNKNOWN_APP_ID),
+#ifdef TxJS
+    mIntrospector(JS::UndefinedValue()),
+#endif
     mParentCharsetSource(0)
 {
     mHistoryID = ++gDocshellIDCounter;
@@ -1798,6 +1801,20 @@
         originIsFile && targetIsFile;
 }
 
+#ifdef TxJS
+NS_IMETHODIMP
+nsDocShell::GetIntrospector(JS::Value *aIntrospector) {
+    *aIntrospector = mIntrospector;
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsDocShell::SetIntrospector(const JS::Value & aIntrospector) {
+    mIntrospector = aIntrospector;
+    return NS_OK;
+}
+#endif
+
 NS_IMETHODIMP
 nsDocShell::GetEldestPresContext(nsPresContext** aPresContext)
 {
@@ -9720,6 +9737,10 @@
 
     uriLoader = do_GetService(NS_URI_LOADER_CONTRACTID, &rv);
     if (NS_FAILED(rv)) return rv;
+#ifdef TxJS
+    JS::Value ispect;
+    GetIntrospector(&ispect);
+#endif
 
     nsLoadFlags loadFlags = mDefaultLoadFlags;
     if (aFirstParty) {
@@ -10014,6 +10035,9 @@
         // Allow execution against our context if the principals match
         scriptChannel->
             SetExecutionPolicy(nsIScriptChannel::EXECUTE_NORMAL);
+#ifdef TxJS
+        scriptChannel->SetIntrospector(ispect);
+#endif
     }
 
     if (aIsNewWindowTarget) {
@@ -10032,7 +10056,7 @@
             timedChannel->SetInitiatorType(NS_LITERAL_STRING("subdocument"));
         }
     }
-
+    static_cast<nsURILoader*>(uriLoader.get())->mIntrospector = ispect;
     rv = DoChannelLoad(channel, uriLoader, aBypassClassifier);
 
     //

