--- ff-orig/mozilla-31.0/js/src/moz.build	2014-07-16 20:45:20.000000000 -0500
+++ ff-dev/mozilla-31.0/js/src/moz.build	2015-01-25 19:38:33.158625500 -0600
@@ -137,6 +137,9 @@
     'jsgc.cpp',
     'jsinfer.cpp',
     'jsiter.cpp',
+    # TxJS begin
+    'jsjam.cpp',
+    # TxJS end
     'jsnativestack.cpp',
     'jsnum.cpp',
     'jsobj.cpp',
@@ -149,6 +152,11 @@
     'jsreflect.cpp',
     'jsscript.cpp',
     'jsstr.cpp',
+    # TxJS begin
+    'jstransaction.cpp',
+    'jstx.cpp',
+    'jstxaction.cpp',
+    # TxJS end
     'jswatchpoint.cpp',
     'jsweakmap.cpp',
     'jsworkers.cpp',

