--- ff-orig/mozilla-31.0/js/src/jsarray.h	2014-07-16 20:45:20.000000000 -0500
+++ ff-dev/mozilla-31.0/js/src/jsarray.h	2015-01-25 19:38:32.794625511 -0600
@@ -11,6 +11,9 @@
 
 #include "jsobj.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 namespace js {
 /* 2^32-2, inclusive */
@@ -164,4 +167,11 @@
 bool
 js_Array(JSContext *cx, unsigned argc, js::Value *vp);
 
+#ifdef TxJS
+extern bool 
+js_SetArrayElement(JSContext *cx, js::HandleObject obj, double index, js::HandleValue vp);
+
+extern bool 
+js_GetArrayElement(JSContext *cx, js::HandleObject obj, double index, bool *hole, js::MutableHandleValue vp);
+#endif
 #endif /* jsarray_h */

