--- ff-orig/mozilla-31.0/js/src/jsatom.cpp	2014-07-16 20:45:20.000000000 -0500
+++ ff-dev/mozilla-31.0/js/src/jsatom.cpp	2015-01-25 19:38:32.810625511 -0600
@@ -86,6 +86,9 @@
 const char js_in_str[]              = "in";
 const char js_instanceof_str[]      = "instanceof";
 const char js_interface_str[]       = "interface";
+#ifdef TxJS
+const char js_introspect_str[]      = "introspect";
+#endif
 const char js_new_str[]             = "new";
 const char js_package_str[]         = "package";
 const char js_private_str[]         = "private";

