--- ff-orig/mozilla-31.0/js/src/shell/js.cpp	2014-07-16 20:45:21.000000000 -0500
+++ ff-dev/mozilla-31.0/js/src/shell/js.cpp	2015-01-25 19:38:33.198625499 -0600
@@ -55,6 +55,10 @@
 #include "jswrapper.h"
 #include "prmjtime.h"
 
+// TxJS begin
+#include "jsversion.h"
+// TxJS end
+
 #include "builtin/TestingFunctions.h"
 #include "frontend/Parser.h"
 #include "jit/arm/Simulator-arm.h"
@@ -2117,7 +2121,12 @@
 JS_STATIC_ASSERT(JSTRY_FINALLY == 1);
 JS_STATIC_ASSERT(JSTRY_ITER == 2);
 
+#ifdef TxJS
+JS_STATIC_ASSERT(JSTRY_TX == 3);
+static const char* const TryNoteNames[] = { "catch", "finally", "iter", "tx", "loop" };
+#else
 static const char* const TryNoteNames[] = { "catch", "finally", "iter", "loop" };
+#endif
 
 static bool
 TryNotes(JSContext *cx, HandleScript script, Sprinter *sp)
@@ -4517,7 +4526,6 @@
     JS_FN_HELP("readline", ReadLine, 0, 0,
 "readline()",
 "  Read a single line from stdin."),
-
     JS_FN_HELP("print", Print, 0, 0,
 "print([exp ...])",
 "  Evaluate and print expressions to stdout."),

