--- ff-orig/mozilla-31.0/parser/html/nsHtml5StreamParser.cpp	2014-07-16 20:45:36.000000000 -0500
+++ ff-dev/mozilla-31.0/parser/html/nsHtml5StreamParser.cpp	2015-01-25 19:38:33.402625493 -0600
@@ -165,6 +165,9 @@
   , mLoadFlusher(new nsHtml5LoadFlusher(aExecutor))
   , mFlushTimer(do_CreateInstance("@mozilla.org/timer;1"))
   , mMode(aMode)
+#ifdef TxJS
+  , mIntrospector(JS::UndefinedValue())
+#endif
 {
   NS_ASSERTION(NS_IsMainThread(), "Wrong thread!");
   mFlushTimer->SetTarget(mThread);
@@ -886,6 +889,9 @@
   mTreeBuilder->SetPreventScriptExecution(!((mMode == NORMAL) &&
                                             scriptingEnabled));
   mTokenizer->start();
+#ifdef TxJS
+  mExecutor->mIntrospector = mIntrospector;
+#endif
   mExecutor->Start();
   mExecutor->StartReadingFromStage();
 

