--- ff-orig/mozilla-25.0/js/src/jsatom.cpp	2013-10-25 17:27:23.000000000 -0500
+++ ff-dev/mozilla-25.0/js/src/jsatom.cpp	2015-01-24 13:25:16.473806531 -0600
@@ -86,6 +86,9 @@
 const char js_in_str[]              = "in";
 const char js_instanceof_str[]      = "instanceof";
 const char js_interface_str[]       = "interface";
+#ifdef TxJS
+const char js_introspect_str[]      = "introspect";
+#endif
 const char js_let_str[]             = "let";
 const char js_new_str[]             = "new";
 const char js_package_str[]         = "package";

