--- ff-orig/mozilla-25.0/js/src/jsscript.h	2013-10-25 17:27:23.000000000 -0500
+++ ff-dev/mozilla-25.0/js/src/jsscript.h	2015-01-24 13:05:15.309841651 -0600
@@ -54,6 +54,9 @@
     JSTRY_CATCH,
     JSTRY_FINALLY,
     JSTRY_ITER,
+#ifdef TxJS
+    JSTRY_TX,
+#endif
     JSTRY_LOOP
 } JSTryNoteKind;
 
@@ -439,6 +442,16 @@
     /* Persistent type information retained across GCs. */
     js::types::TypeScript *types;
 
+#ifdef TxJS
+  private:
+    js::Value introspector_;
+  public:
+    inline const js::Value getIntrospector() { return introspector_; }
+    inline void setIntrospector(const js::Value &ispect) {
+        introspector_ = ispect;
+    }
+#endif
+
   private:
     js::HeapPtrObject sourceObject_; /* source code object */
     js::HeapPtrFunction function_;

