--- ff-orig/mozilla-25.0/js/src/frontend/ParseNode.h	2013-10-25 17:27:22.000000000 -0500
+++ ff-dev/mozilla-25.0/js/src/frontend/ParseNode.h	2015-01-24 13:00:50.217849401 -0600
@@ -199,6 +199,9 @@
 #define EMIT_ENUM(name) PNK_##name,
     FOR_EACH_PARSE_NODE_KIND(EMIT_ENUM)
 #undef EMIT_ENUM
+#ifdef TxJS
+    PNK_TX,
+#endif
     PNK_LIMIT, /* domain size */
     PNK_BINOP_FIRST = PNK_OR,
     PNK_BINOP_LAST = PNK_MOD,

