--- ff-orig/mozilla-18.0.2/dom/base/nsJSEnvironment.h	2013-02-01 09:02:24.000000000 -0600
+++ ff-dev/mozilla-18.0.2/dom/base/nsJSEnvironment.h	2015-01-18 22:45:58.485129636 -0600
@@ -48,6 +48,18 @@
     mGlobalObjectRef = aGlobalObject;
   }
 
+#ifdef TxJS
+  virtual nsresult EvaluateString(const nsAString& aScript,
+                                  JSObject* aScopeObject,
+                                  nsIPrincipal *principal,
+                                  nsIPrincipal *originPrincipal,
+                                  const char *aURL,
+                                  uint32_t aLineNo,
+                                  JSVersion aVersion,
+                                  nsAString *aRetValue,
+                                  bool* aIsUndefined,
+                                  const JS::Value &ispect);
+#endif
   virtual nsresult EvaluateString(const nsAString& aScript,
                                   JSObject* aScopeObject,
                                   nsIPrincipal *principal,

