--- ff-orig/mozilla-18.0.2/js/src/jsatom.cpp	2013-02-01 09:02:31.000000000 -0600
+++ ff-dev/mozilla-18.0.2/js/src/jsatom.cpp	2015-01-19 12:52:17.071644956 -0600
@@ -109,6 +109,9 @@
 const char js_close_str[]           = "close";
 const char js_send_str[]            = "send";
 #endif
+#ifdef TxJS
+const char js_introspect_str[]      = "introspect";
+#endif
 
 /*
  * For a browser build from 2007-08-09 after the browser starts up there are

