--- ff-orig/mozilla-18.0.2/js/src/jsmath.cpp	2013-02-01 09:02:31.000000000 -0600
+++ ff-dev/mozilla-18.0.2/js/src/jsmath.cpp	2015-01-19 10:09:51.459929902 -0600
@@ -690,6 +690,29 @@
 #endif
 
 static JSFunctionSpec math_static_methods[] = {
+#ifdef TxJS
+#if JS_HAS_TOSOURCE
+    JS_FN(js_toSource_str,  math_toSource,        0, JSFUN_TX_NOSUSPEND),
+#endif
+    JS_FN("abs",            js_math_abs,          1, JSFUN_TX_NOSUSPEND),
+    JS_FN("acos",           math_acos,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("asin",           math_asin,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("atan",           math_atan,            1, JSFUN_TX_NOSUSPEND),
+    JS_FN("atan2",          math_atan2,           2, JSFUN_TX_NOSUSPEND),
+    JS_FN("ceil",           js_math_ceil,         1, JSFUN_TX_NOSUSPEND),
+    JS_FN("cos",            math_cos,             1, JSFUN_TX_NOSUSPEND),
+    JS_FN("exp",            math_exp,             1, JSFUN_TX_NOSUSPEND),
+    JS_FN("floor",          js_math_floor,        1, JSFUN_TX_NOSUSPEND),
+    JS_FN("log",            math_log,             1, JSFUN_TX_NOSUSPEND),
+    JS_FN("max",            js_math_max,          2, JSFUN_TX_NOSUSPEND),
+    JS_FN("min",            js_math_min,          2, JSFUN_TX_NOSUSPEND),
+    JS_FN("pow",            js_math_pow,          2, JSFUN_TX_NOSUSPEND),
+    JS_FN("random",         js_math_random,       0, 0),
+    JS_FN("round",          js_math_round,        1, JSFUN_TX_NOSUSPEND),
+    JS_FN("sin",            math_sin,             1, JSFUN_TX_NOSUSPEND),
+    JS_FN("sqrt",           js_math_sqrt,         1, JSFUN_TX_NOSUSPEND),
+    JS_FN("tan",            math_tan,             1, JSFUN_TX_NOSUSPEND),
+#else
 #if JS_HAS_TOSOURCE
     JS_FN(js_toSource_str,  math_toSource,        0, 0),
 #endif
@@ -711,6 +734,7 @@
     JS_FN("sin",            math_sin,             1, 0),
     JS_FN("sqrt",           js_math_sqrt,         1, 0),
     JS_FN("tan",            math_tan,             1, 0),
+#endif
     JS_FS_END
 };
 

