--- ff-orig/mozilla-18.0.2/js/src/vm/Stack.cpp	2013-02-01 09:02:33.000000000 -0600
+++ ff-dev/mozilla-18.0.2/js/src/vm/Stack.cpp	2015-01-18 23:59:41.817000305 -0600
@@ -85,6 +85,9 @@
     hookData_ = (void *)0xbad;
     annotation_ = (void *)0xbad;
 #endif
+#ifdef TxJS
+    suspendedTx_ = NULL;
+#endif
 
     if (prev && prev->annotation())
         setAnnotation(prev->annotation());
@@ -1063,6 +1066,19 @@
     fp->initCallFrame(cx, *fun, script, args.length(), flags);
     fg->regs_.prepareToRun(*fp, script);
 
+#ifdef TxJS
+    // Save a transaction that is in the process of suspending, so that
+    // we can automatically resume it when this frame returns.
+    TxContext *tx = cx->getSuspendingTx();
+    if (tx) {
+        JS_ASSERT(!fp->suspendedTx());
+        // Advance the transaction state.
+        JS_ASSERT(tx->inState(TX_SUSPEND0 | TX_END0));
+        tx->advanceState();
+        fp->setSuspendedTx(tx);
+    }
+#endif
+
     fg->prevRegs_ = seg_->pushRegs(fg->regs_);
     JS_ASSERT(space().firstUnused() == fg->regs_.sp);
     fg->setPushed(*this);
@@ -1122,6 +1138,18 @@
     fp->initVarsToUndefined();
     efg->regs_.prepareToRun(*fp, script);
 
+#ifdef TxJS
+    // Save a transaction that is in the process of suspending, so that
+    // we can automatically resume it when this frame returns.
+    TxContext *tx = cx->getSuspendingTx();
+    if (tx) {
+        JS_ASSERT(!fp->suspendedTx());
+        JS_ASSERT(tx->inState(TX_SUSPEND0 | TX_END0));
+        tx->advanceState();
+        fp->setSuspendedTx(tx);
+    }
+#endif
+
     /* pushRegs() below links the prev-frame; manually link the prev-call. */
     if (evalInFrame && evalInFrameCalls)
         seg_->pointAtCall(*evalInFrameCalls);
@@ -1171,6 +1199,51 @@
 
     Value *oldend = seg_->end();
 
+#ifdef TxJS
+    /*
+    if (fg.prevRegs_) {
+        StackFrame *fp = fg.prevRegs_->fp();
+        if (fp) {
+            TxContext::clearFrameLocals(cx_, fp);
+            TxContext *tx = fp->suspendedTx();
+            if (tx) {
+                JS_ASSERT(tx->inState(TX_SUSPEND1 | TX_END1));
+                // TX_SUSPEND1|TX_END1 -> TX_RESUME|TX_DONE
+                tx->advanceState();
+                fp->setSuspendedTx(NULL);
+
+                // Perform actions prior to popping based on suspend type.
+                // %%% Necessary?
+                if (tx->inState(TX_RESUME)) {
+                    switch (tx->getSuspend()->type) {
+                      case AX_WRITE:
+                      case AX_CALL:
+                        // Push the return value of the handler frame.
+                        fp->setReturnValue(tx->getReturnValue());
+                        break;
+                      default:
+                        break;
+                    }
+                }
+
+                if (tx && tx->inState(TX_RESUME | TX_DONE)) {
+                    // An exception occurred within a transaction handler.
+                    // %%% Always true here?
+                    // If the transaction was suspended, we need to restart
+                    // it here.
+                    if (tx->inState(TX_RESUME)) {
+                        // TX_RESUME -> TX_RUNNING
+                        tx->advanceState();
+                    }
+                    // %%% Only do this for TX_RESUME case also?
+                    //tx = cx_->setTxContext(tx);
+                }
+            }
+        }
+    }
+    */
+#endif
+
     seg_->popRegs(fg.prevRegs_);
     if (fg.pushedSeg_)
         popSegment();
@@ -1284,6 +1357,38 @@
 StackIter::popFrame()
 {
     StackFrame *oldfp = fp_;
+#ifdef TxJS
+    /*
+    if (maybecx_) {
+        TxContext::clearFrameLocals(maybecx_, oldfp);
+    }
+    TxContext *tx = oldfp->suspendedTx();
+    if (tx && tx->inState(TX_SUSPEND1)) {
+        //tx->print();
+        JS_ASSERT(maybecx_);
+        // TX_SUSPEND1|TX_END1 -> TX_RESUME|TX_DONE
+        tx->advanceState();
+        oldfp->setSuspendedTx(NULL);
+
+        // Perform actions prior to popping based on suspend type.
+        // %%% Necessary?
+        switch (tx->getSuspend()->type) {
+          case AX_WRITE:
+          case AX_CALL:
+            // Push the return value of the handler frame.
+            oldfp->setReturnValue(tx->getReturnValue());
+            break;
+          default:
+            break;
+        }
+
+        // TX_RESUME -> TX_RUNNING
+        tx->advanceState();
+        // %%% Only do this for TX_RESUME case also?
+        //tx = maybecx_->setTxContext(tx);
+    }
+    */
+#endif
     JS_ASSERT(seg_->contains(oldfp));
     fp_ = fp_->prev();
 
@@ -1300,6 +1405,9 @@
 void
 StackIter::popCall()
 {
+#ifdef TxJS
+    JS_ASSERT(!fp_ || !fp_->suspendedTx());
+#endif
     DebugOnly<CallArgsList*> oldCall = calls_;
     JS_ASSERT(seg_->contains(oldCall));
     calls_ = calls_->prev();

