--- ff-orig/mozilla-18.0.2/js/src/frontend/TokenStream.h	2013-02-01 09:02:30.000000000 -0600
+++ ff-dev/mozilla-18.0.2/js/src/frontend/TokenStream.h	2015-01-19 14:36:02.895462924 -0600
@@ -18,6 +18,9 @@
 #include "jsopcode.h"
 #include "jsprvtd.h"
 #include "jspubtd.h"
+#ifdef TxJS
+#include "jstx.h"
+#endif
 
 #include "js/Vector.h"
 
@@ -153,7 +156,9 @@
     TOK_DIVASSIGN,
     TOK_MODASSIGN,
     TOK_ASSIGNMENT_LAST = TOK_MODASSIGN,
-
+#ifdef TxJS
+    TOK_TX,
+#endif
     TOK_LIMIT                      /* domain size */
 };
 

