--- ff-orig/mozilla-18.0.2/js/src/jsprototypes.h	2013-02-01 09:02:31.000000000 -0600
+++ ff-dev/mozilla-18.0.2/js/src/jsprototypes.h	2015-01-19 16:05:25.175306140 -0600
@@ -60,5 +60,7 @@
     macro(Set,                   38,     js_InitSetClass) \
     macro(DataView,              39,     js_InitTypedArrayClasses) \
     macro(ParallelArray,         40,     js_InitParallelArrayClass) \
+    macro(Tx,                    41,     js_InitTxClass) \
+    macro(JAM,                   42,     js_InitJAMClass) \
 
 #endif /* jsprototypes_h___ */

